<!DOCTYPE html>
<html lang="en"><head>
<script src="03-biostatistics_files/libs/clipboard/clipboard.min.js"></script>
<script src="03-biostatistics_files/libs/quarto-html/tabby.min.js"></script>
<script src="03-biostatistics_files/libs/quarto-html/popper.min.js"></script>
<script src="03-biostatistics_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="03-biostatistics_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="03-biostatistics_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="03-biostatistics_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="03-biostatistics_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="03-biostatistics_files/libs/quarto-contrib/jquery-3.6.3/jquery-3.6.3.min.js"></script>
<script src="03-biostatistics_files/libs/quarto-contrib/code-fullscreen-1.0.0/code-fullscreen.js" defer="true"></script>
<link href="03-biostatistics_files/libs/quarto-contrib/code-fullscreen-1.0.0/code-fullscreen.css" rel="stylesheet"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
  <meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.450">

  <meta name="author" content="Filippo Gambarota">
  <meta name="author" content="Gianmarco Alto√®">
  <title>Exploring Replicability in Biostatistics</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="03-biostatistics_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="03-biostatistics_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="03-biostatistics_files/libs/revealjs/dist/theme/quarto.css">
  <link href="03-biostatistics_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="03-biostatistics_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="03-biostatistics_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="03-biostatistics_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
  
  <script src="03-biostatistics_files/libs/kePrint-0.0.1/kePrint.js"></script>
  <link href="03-biostatistics_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">
  <link href="03-biostatistics_files/libs/font-awesome-5.13.0/css/all.min.css" rel="stylesheet">
  <link href="03-biostatistics_files/libs/font-awesome-5.13.0/css/v4-shims.min.css" rel="stylesheet">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Exploring Replicability in Biostatistics</h1>
  <p class="subtitle">Replicability Crisis in Science?</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Filippo Gambarota <a href="https://orcid.org/0000-0002-6666-1747" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a>
</div>
<div class="quarto-title-author-email">
<a href="mailto:filippo.gambarota@unipd.it">filippo.gambarota@unipd.it</a>
</div>
</div>
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Gianmarco Alto√® <a href="https://orcid.org/0000-0003-1154-9528" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a>
</div>
<div class="quarto-title-author-email">
<a href="mailto:gianmarco.altoe@unipd.it">gianmarco.altoe@unipd.it</a>
</div>
</div>
</div>

  <p class="date">18-22 September 2023</p>
</section>
<section class="notation-slide" ;="" id="notation-slide">
  <h2>General info</h2>
<br>
<br>
  <p>
    </p><ul>
      <li>
        Slides marked with üí¨ are <b>EXTRA</b> slides with useful but not fundamental information
      </li>
      <li>
        Slides marked with ‚ùì are <b>QUESTIONS</b> slide
      </li>
      <li>
        <div class="cell" data-layout-align="center">
          <details>
          <summary>This is a collapsed code-block, you can open it and see the code</summary>
          </details>
        </div>
      </li>
    </ul>
  <p></p>
</section>

<section id="setup-r" class="slide level2">
<h2>Setup R</h2>
<h3 id="packages">Packages</h3>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># for data manipulation</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(curatedOvarianData) <span class="co"># workshop data</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(pROC)</span>
<span id="cb1-4"><a href="#cb1-4"></a>devtools<span class="sc">::</span><span class="fu">load_all</span>() <span class="co"># load all functions for the workshop</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<h3 id="loading-data">Loading Data</h3>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>datl <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"03-biostatistics"</span>, <span class="st">"objects"</span>, <span class="st">"gene_data_clean.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section>
<section id="reproducibility-in-biostatistics" class="title-slide slide level1 center">
<h1>Reproducibility in Biostatistics</h1>

</section>
<section id="reproducibility-in-biostatistics-1" class="slide level2">
<h2>Reproducibility in Biostatistics</h2>
<ul>
<li><p>In <em>Biostatistics</em> is common to have a similar experimental setup (i.e., same genes and variables), repeated across multiple studies. Replicability in this framework refer to how consistent is a variable (e.g., gene expression) in predicting a certain outcome across different dataset.</p></li>
<li><p>Recent research suggest the importance of cross-study training of prediction models to improve the overall predictive performance and replicability <span class="citation" data-cites="Patil2018-hm Parmigiani2023-dr Masoero2023-wb">(<a href="#/references" role="doc-biblioref" onclick="">Patil and Parmigiani 2018</a>; <a href="#/references" role="doc-biblioref" onclick="">Parmigiani 2023</a>; <a href="#/references" role="doc-biblioref" onclick="">Masoero et al. 2023</a>)</span></p></li>
<li><p>We will try to assess how a certain variable consistently (or not) predict a certain outcome using a binary classification and meta-analytic approach</p></li>
</ul>
</section></section>
<section>
<section id="background" class="title-slide slide level1 section center">
<h1>Background</h1>

</section>
<section id="background-1" class="slide level2">
<h2>Background</h2>
<p>For this lecture we will use:</p>
<ul>
<li>Gene expression data</li>
<li>Analysis of <strong>contingency tables</strong></li>
<li>Binary classifiers</li>
<li><strong>Receiver Operating Characteristic</strong> (ROC) curves</li>
<li>The <strong>Youden index</strong> to analyze ROC curves</li>
</ul>
</section>
<section id="genes-expression-data" class="slide level2">
<h2>Genes expression data</h2>
<p>We are going to use the <code>curatedOvarianData</code> dataset <span class="citation" data-cites="Waldron2014-yv">(see <a href="#/references" role="doc-biblioref" onclick="">Waldron et al. 2014</a> for details)</span>. Ganzfried <span class="citation" data-cites="Ganzfried2013-ut">(<a href="#/references" role="doc-biblioref" onclick="">2013</a>)</span> provide a detailed description of the dataset structure.</p>
<div class="cell" data-layout-align="center">
<iframe src="https://bioconductor.org/packages/release/data/experiment/html/curatedOvarianData.html" width="960" height="400px" data-external="1">
</iframe>
</div>
</section>
<section id="data-structure" class="slide level2">
<h2>Data structure</h2>
<p>Gene <strong>Expression</strong> data are very complex in terms of acquisition, pre-processing and organization. For the purpose of this workshop we will transform the gene-phenotype database in a standard 2d dataframe using the <code>get_bio_data()</code> function. The majority of packages that we are going to use works better with this data structure.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>get_bio_data <span class="ot">&lt;-</span> <span class="cf">function</span>(data, gene, phenotype){</span>
<span id="cb3-2"><a href="#cb3-2"></a>  phend <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Biobase<span class="sc">::</span><span class="fu">pData</span>(data)[, phenotype])</span>
<span id="cb3-3"><a href="#cb3-3"></a>  gened <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Biobase<span class="sc">::</span><span class="fu">exprs</span>(data)[gene, ])</span>
<span id="cb3-4"><a href="#cb3-4"></a>  <span class="cf">if</span>(<span class="fu">length</span>(gene) <span class="sc">!=</span> <span class="dv">1</span>){</span>
<span id="cb3-5"><a href="#cb3-5"></a>    gened <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(gened))</span>
<span id="cb3-6"><a href="#cb3-6"></a>  }</span>
<span id="cb3-7"><a href="#cb3-7"></a>  out <span class="ot">&lt;-</span> <span class="fu">cbind</span>(phend, gened)</span>
<span id="cb3-8"><a href="#cb3-8"></a>  <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">c</span>(phenotype, gene)</span>
<span id="cb3-9"><a href="#cb3-9"></a>  <span class="fu">rownames</span>(out) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-10"><a href="#cb3-10"></a>  <span class="fu">return</span>(out)</span>
<span id="cb3-11"><a href="#cb3-11"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-structure-1" class="slide level2">
<h2>Data structure</h2>
<p>All the datasets are available using the <code>data(dataset, package = "curatedOvarianData")</code>. To see a list of available datasets (when the package is loaded using <code>library()</code> the <code>package =</code> argument can be omitted):</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="fu">data</span>(<span class="at">package =</span> <span class="st">"curatedOvarianData"</span>)<span class="sc">$</span>result[, <span class="st">"Item"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "E.MTAB.386_eset"         "GSE12418_eset"          
 [3] "GSE12470_eset"           "GSE13876_eset"          
 [5] "GSE14764_eset"           "GSE17260_eset"          
 [7] "GSE18520_eset"           "GSE19829.GPL570_eset"   
 [9] "GSE19829.GPL8300_eset"   "GSE20565_eset"          
[11] "GSE2109_eset"            "GSE26193_eset"          
[13] "GSE26712_eset"           "GSE30009_eset"          
[15] "GSE30161_eset"           "GSE32062.GPL6480_eset"  
[17] "GSE32063_eset"           "GSE44104_eset"          
[19] "GSE49997_eset"           "GSE51088_eset"          
[21] "GSE6008_eset"            "GSE6822_eset"           
[23] "GSE8842_eset"            "GSE9891_eset"           
[25] "PMID15897565_eset"       "PMID17290060_eset"      
[27] "PMID19318476_eset"       "TCGA.RNASeqV2_eset"     
[29] "TCGA.mirna.8x15kv2_eset" "TCGA_eset"              </code></pre>
</div>
</div>
</section>
<section id="data-structure-2" class="slide level2">
<h2>Data structure</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code>                      Item
1          E.MTAB.386_eset
2            GSE12418_eset
3            GSE12470_eset
4            GSE13876_eset
5            GSE14764_eset
6            GSE17260_eset
7            GSE18520_eset
8     GSE19829.GPL570_eset
9    GSE19829.GPL8300_eset
10           GSE20565_eset
11            GSE2109_eset
12           GSE26193_eset
13           GSE26712_eset
14           GSE30009_eset
15           GSE30161_eset
16   GSE32062.GPL6480_eset
17           GSE32063_eset
18           GSE44104_eset
19           GSE49997_eset
20           GSE51088_eset
21            GSE6008_eset
22            GSE6822_eset
23            GSE8842_eset
24            GSE9891_eset
25       PMID15897565_eset
26       PMID17290060_eset
27       PMID19318476_eset
28      TCGA.RNASeqV2_eset
29 TCGA.mirna.8x15kv2_eset
30               TCGA_eset
                                                                                                                                                                       Title
1                                                                  Angiogenic mRNA and microRNA gene expression signature predicts a novel subtype of serous ovarian cancer.
2                                                                     Expression analysis of stage III serous ovarian adenocarcinoma distinguishes a sub-group of survivors.
3                  Gene expression profiling of advanced-stage serous ovarian cancers distinguishes novel subclasses and implicates ZEB2 in tumor progression and prognosis.
4                                                                                           Survival-related profile, pathways, and transcription factors in ovarian cancer.
5                                                                  A prognostic gene expression index in ovarian cancer - validation across different independent data sets.
6                                                   Gene expression profile for predicting survival in advanced-stage serous ovarian cancer across two independent datasets.
7                                    A gene signature predictive for outcome in advanced ovarian cancer identifies a survival factor: microfibril-associated glycoprotein 2.
8                       Gene expression profile of BRCAness that correlates with responsiveness to chemotherapy and with outcome in patients with epithelial ovarian cancer.
9                       Gene expression profile of BRCAness that correlates with responsiveness to chemotherapy and with outcome in patients with epithelial ovarian cancer.
10 A genomic and transcriptomic approach for a differential diagnosis between primary and secondary ovarian carcinomas in patients with a previous history of breast cancer.
11                                                                                                                                       IGC EXpression Project for Oncology
12                                                                               miR-141 and miR-200a act on ovarian tumorigenesis by controlling oxidative stress response.
13                                                                           A gene signature predicting for survival in suboptimally debulked patients with ovarian cancer.
14                                                   Multidrug resistance-linked gene signature predicts overall survival of patients with primary ovarian serous carcinoma.
15                                  Multi-gene expression predictors of single drug responses to adjuvant chemotherapy in ovarian carcinoma: predicting platinum resistance.
16                              High-risk ovarian cancer based on 126-gene expression signature is uniquely characterized by downregulation of antigen presentation pathway.
17                              High-risk ovarian cancer based on 126-gene expression signature is uniquely characterized by downregulation of antigen presentation pathway.
18                                                                                  COL11A1 promotes tumor progression and predicts poor clinical outcome in ovarian cancer.
19                                                              Validating the impact of a molecular subtype in ovarian cancer on outcomes: a study of the OVCAD Consortium.
20                                                                     POSTN/TGFBI-associated stromal signature predicts poor prognosis in serous epithelial ovarian cancer.
21                                              Lysophosphatidic acid-induced transcriptional profile represents serous epithelial ovarian carcinoma and worsened prognosis.
22                                                                                                                                   Classification of ovarian tumor samples
23                                                                                                                Analysis of gene expression in early-stage ovarian cancer.
24                                                                            Novel molecular subtypes of serous and endometrioid ovarian cancer linked to clinical outcome.
25                                                                Patterns of gene expression that characterize long-term survival in advanced stage serous ovarian cancers.
26                                                          An integrated genomic-based approach to individualized treatment of patients with advanced-stage ovarian cancer.
27                                                                 Microarray analysis of early stage serous ovarian cancers shows profiles predictive of favorable outcome.
28                                                                                                                         Integrated genomic analyses of ovarian carcinoma.
29                                                                                                                         Integrated genomic analyses of ovarian carcinoma.
30                                                                                                                         Integrated genomic analyses of ovarian carcinoma.</code></pre>
</div>
</div>
</section>
<section id="data-structure-3" class="slide level2">
<h2>Data structure</h2>
<p>For example, we want to extract the <code>ARHGAP5</code> and <code>ZNF487</code> <em>genes</em> and <code>debulking</code> and <code>tumorstage</code> <em>phenotype</em> variables.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">data</span>(GSE32063_eset) <span class="co"># load the dataset</span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="fu">get_bio_data</span>(GSE32063_eset, <span class="fu">c</span>(<span class="st">"ARHGAP5"</span>, <span class="st">"ZNF487"</span>), <span class="fu">c</span>(<span class="st">"debulking"</span>, <span class="st">"tumorstage"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    debulking tumorstage    ARHGAP5     ZNF487
1  suboptimal          3 -1.3788404 -1.4693060
2     optimal          3 -0.7029762 -2.1866198
3  suboptimal          3 -1.1906614 -1.5242004
4  suboptimal          3 -1.5408096 -1.9636803
5  suboptimal          3 -1.8204670 -1.6253910
6     optimal          3 -0.1023159 -0.9768720
7  suboptimal          4 -1.4978333 -2.8806515
8     optimal          4 -0.3169098 -0.9902863
9     optimal          3 -0.9621868 -1.2781801
10 suboptimal          3 -2.6246548 -2.3127084
11 suboptimal          3 -1.1208530 -1.3294687
12 suboptimal          3 -1.6579170 -3.5191045
13 suboptimal          4 -1.7063255 -1.0643783
14    optimal          3 -1.2869992 -1.2185640
15 suboptimal          3 -2.5078154 -1.6630969
16 suboptimal          4 -1.0130329 -1.6677127
17 suboptimal          3 -2.3309345 -2.7753735
18 suboptimal          3 -2.4601330 -1.0582757
19    optimal          3 -1.7182579 -1.4094768
20    optimal          3 -1.3721395 -1.7538829
21    optimal          4 -1.0870414 -1.6791105
22    optimal          3 -0.7623720 -1.4829173
23 suboptimal          3 -1.1356659 -2.1259017
24    optimal          3 -1.3679762 -0.1064415
25 suboptimal          4 -1.1441503 -2.5149746
26    optimal          3 -1.8230276 -0.3728800
27 suboptimal          3 -1.2220440 -2.3130960
28    optimal          3 -0.7620802 -1.6277437
29    optimal          3 -0.1756640 -1.4368038
30 suboptimal          3 -0.3513031 -1.5798197
31    optimal          3 -1.1255894 -1.3647880
32    optimal          3 -1.2265711 -1.5709486
33    optimal          3  0.4664946 -1.0424862
34 suboptimal          4 -1.8461108 -2.3798810
35    optimal          3 -1.7979369 -0.7189474
36 suboptimal          3 -1.6521063 -1.7661219
37    optimal          3 -2.3519936 -0.6084919
38    optimal          3 -2.1397367 -0.6968164
39 suboptimal          4 -1.6583776 -2.9544182
40 suboptimal          4 -1.6778574 -3.0424485</code></pre>
</div>
</div>
</section>
<section id="data-structure-4" class="slide level2">
<h2>Data structure</h2>
<p>We performed a little bit of pre-processing to have a more clean and minimal dataset. The pre-processing are described into the <code>03-biostatistics/scripts/gene-data-preprocessing.R</code>:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">library</span>(curatedOvarianData)</span>
<span id="cb9-2"><a href="#cb9-2"></a>devtools<span class="sc">::</span><span class="fu">load_all</span>()</span>
<span id="cb9-3"><a href="#cb9-3"></a></span>
<span id="cb9-4"><a href="#cb9-4"></a>listdata <span class="ot">&lt;-</span> <span class="fu">data</span>(<span class="at">package =</span> <span class="st">"curatedOvarianData"</span>)<span class="sc">$</span>results[, <span class="st">"Item"</span>]</span>
<span id="cb9-5"><a href="#cb9-5"></a>datl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(listdata, getdata, <span class="at">package =</span> <span class="st">"curatedOvarianData"</span>)</span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="fu">names</span>(datl) <span class="ot">&lt;-</span> listdata</span>
<span id="cb9-7"><a href="#cb9-7"></a></span>
<span id="cb9-8"><a href="#cb9-8"></a><span class="co"># we need the debulking as phenotype, lets see which data have it</span></span>
<span id="cb9-9"><a href="#cb9-9"></a></span>
<span id="cb9-10"><a href="#cb9-10"></a>has_debulking <span class="ot">&lt;-</span> <span class="fu">sapply</span>(datl, <span class="cf">function</span>(d) <span class="fu">any</span>(<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">pData</span>(d)[, <span class="st">"debulking"</span>])))</span>
<span id="cb9-11"><a href="#cb9-11"></a>datl_debulking <span class="ot">&lt;-</span> datl[has_debulking]</span>
<span id="cb9-12"><a href="#cb9-12"></a></span>
<span id="cb9-13"><a href="#cb9-13"></a><span class="co"># let's see how many NAs and remove the dataset with an high percentage</span></span>
<span id="cb9-14"><a href="#cb9-14"></a></span>
<span id="cb9-15"><a href="#cb9-15"></a>pNA <span class="ot">&lt;-</span> <span class="fu">sapply</span>(datl_debulking, <span class="cf">function</span>(x) <span class="fu">mean</span>(<span class="fu">is.na</span>(<span class="fu">pData</span>(x)[, <span class="st">"debulking"</span>])))</span>
<span id="cb9-16"><a href="#cb9-16"></a>ncases <span class="ot">&lt;-</span> <span class="fu">sapply</span>(datl_debulking, <span class="cf">function</span>(x) <span class="fu">length</span>(<span class="fu">pData</span>(x)[, <span class="st">"debulking"</span>]))</span>
<span id="cb9-17"><a href="#cb9-17"></a>missing <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pNA, ncases)</span>
<span id="cb9-18"><a href="#cb9-18"></a></span>
<span id="cb9-19"><a href="#cb9-19"></a><span class="co"># remove dataset with more than 20% of NA on debulking</span></span>
<span id="cb9-20"><a href="#cb9-20"></a>datl_debulking <span class="ot">&lt;-</span> datl_debulking[pNA <span class="sc">&lt;</span> <span class="fl">0.2</span>]</span>
<span id="cb9-21"><a href="#cb9-21"></a></span>
<span id="cb9-22"><a href="#cb9-22"></a><span class="co"># let's explore the genes, we keep the genes that are in common with all</span></span>
<span id="cb9-23"><a href="#cb9-23"></a><span class="co"># datasets. One dataset the "TCGA.mirna.8x15kv2_eset" dataset has some genes that are</span></span>
<span id="cb9-24"><a href="#cb9-24"></a><span class="co"># completely different from the others, we remove it.</span></span>
<span id="cb9-25"><a href="#cb9-25"></a></span>
<span id="cb9-26"><a href="#cb9-26"></a><span class="co"># get_gene_names(datl_debulking$TCGA.mirna.8x15kv2_eset)</span></span>
<span id="cb9-27"><a href="#cb9-27"></a>datl_debulking <span class="ot">&lt;-</span> datl_debulking[<span class="fu">names</span>(datl_debulking) <span class="sc">!=</span> <span class="st">"TCGA.mirna.8x15kv2_eset"</span>]</span>
<span id="cb9-28"><a href="#cb9-28"></a></span>
<span id="cb9-29"><a href="#cb9-29"></a><span class="co"># now we intersect all the gene names to keep only the common ones</span></span>
<span id="cb9-30"><a href="#cb9-30"></a>genes <span class="ot">&lt;-</span> <span class="fu">lapply</span>(datl_debulking, get_gene_names)</span>
<span id="cb9-31"><a href="#cb9-31"></a>common_genes <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">Reduce</span>(base<span class="sc">::</span>intersect, genes)</span>
<span id="cb9-32"><a href="#cb9-32"></a></span>
<span id="cb9-33"><a href="#cb9-33"></a><span class="co"># converting into dataframe with selected </span></span>
<span id="cb9-34"><a href="#cb9-34"></a>datl_debulking_df <span class="ot">&lt;-</span> <span class="fu">lapply</span>(datl_debulking, get_bio_data, common_genes, <span class="st">"debulking"</span>)</span>
<span id="cb9-35"><a href="#cb9-35"></a></span>
<span id="cb9-36"><a href="#cb9-36"></a><span class="co"># saving</span></span>
<span id="cb9-37"><a href="#cb9-37"></a><span class="fu">saveRDS</span>(datl_debulking_df, <span class="st">"03-biostatistics/objects/gene_data_clean.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-structure-5" class="slide level2">
<h2>Data structure</h2>
<p>The pre-processing create a dataframe with datasets having both the <code>debulking</code> variable and all genes in common. In this way we can analyze and compare genes among all datasets.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>datl <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"03-biostatistics"</span>, <span class="st">"objects"</span>, <span class="st">"gene_data_clean.rds"</span>))</span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="fu">str</span>(datl, <span class="at">max.level =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 15
 $ E.MTAB.386_eset      :'data.frame':  129 obs. of  220 variables:
 $ GSE12418_eset        :'data.frame':  54 obs. of  220 variables:
 $ GSE17260_eset        :'data.frame':  110 obs. of  220 variables:
 $ GSE26712_eset        :'data.frame':  195 obs. of  220 variables:
 $ GSE30009_eset        :'data.frame':  103 obs. of  220 variables:
 $ GSE30161_eset        :'data.frame':  58 obs. of  220 variables:
 $ GSE32062.GPL6480_eset:'data.frame':  260 obs. of  220 variables:
 $ GSE32063_eset        :'data.frame':  40 obs. of  220 variables:
 $ GSE49997_eset        :'data.frame':  204 obs. of  220 variables:
 $ GSE9891_eset         :'data.frame':  285 obs. of  220 variables:
 $ PMID15897565_eset    :'data.frame':  63 obs. of  220 variables:
 $ PMID17290060_eset    :'data.frame':  117 obs. of  220 variables:
 $ PMID19318476_eset    :'data.frame':  42 obs. of  220 variables:
 $ TCGA.RNASeqV2_eset   :'data.frame':  261 obs. of  220 variables:
 $ TCGA_eset            :'data.frame':  578 obs. of  220 variables:</code></pre>
</div>
</div>
</section>
<section id="data-structure-6" class="slide level2">
<h2>Data structure</h2>
<p>I prefer always working (if possible) with a simple dataframe:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>dat <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(datl, <span class="at">.id =</span> <span class="st">"dataset"</span>)</span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="fu">head</span>(dat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          dataset  debulking     ABCA1    ABCA2    ABCA3    ABCA4    ABCA5
1 E.MTAB.386_eset suboptimal 10.839395 6.845599 8.962037 7.735899 6.911055
2 E.MTAB.386_eset    optimal 11.227465 7.522329 8.647420 7.257863 7.157317
3 E.MTAB.386_eset    optimal 11.375979 7.195961 8.124816 8.797894 7.076414
4 E.MTAB.386_eset    optimal  9.716625 8.253187 7.494501 7.685826 6.926470
5 E.MTAB.386_eset suboptimal 11.647426 8.300361 7.816192 7.241782 7.062383
6 E.MTAB.386_eset    optimal 10.291002 7.487264 7.788690 9.890838 7.227646
     ABCA8    ABCB1    ABCB4    ABCB7    ABCB9    ABCC1    ABCC10    ABCC4
1 8.880053 7.742924 6.858119 10.26362 7.233300 12.47511 10.621616 7.867795
2 6.705939 7.290139 6.896026 11.16421 8.294693 10.96171  9.283463 7.702067
3 7.542021 7.331410 6.729391 10.86380 7.547010 10.20072  8.884857 7.847243
4 7.827795 6.655951 6.708140 11.70487 8.637981 12.33818  9.458241 8.074573
5 6.796258 7.791991 7.488330 11.05764 8.941183 12.19990  9.561413 7.654644
6 7.586522 7.248623 6.837272 11.19498 7.516927 10.58948  9.032422 8.400274
     ABCC5     ABCC8    ABCD2    ABCD3    ABCE1    ABCF2    ABCF3     ABCG1
1 8.971456 10.368624 6.569425 6.864870 8.765213 8.405781 8.355384 12.119868
2 8.924245  7.882769 7.349124 7.035186 9.480066 9.613636 8.567225  9.533904
3 9.317958  7.240344 6.650674 6.865343 8.716176 9.322715 7.889017  9.246897
4 8.532776 10.257129 6.611022 6.761457 9.122343 9.282883 8.822519 11.375979
5 9.674851 11.258042 6.614314 6.744717 8.964149 6.764569 7.975910 12.136104
6 9.402390  7.345348 6.610154 6.916254 9.292160 9.316871 9.647487  8.672094
      ABCG4    ANXA1    ANXA4     APAF1      APC    APEX1     APEX2      APOE
1 12.644486 7.758441 6.780904  9.429752 6.659251 11.26058  9.964905 12.939256
2  8.937479 6.759182 6.843355  9.716625 6.541244 11.38137 10.379333 13.526659
3  9.032422 7.987442 6.792590  9.872901 6.756587 11.98111 10.470689  9.321591
4 11.081676 7.457984 6.699457  9.522337 6.649057 13.52122 10.019142 10.797752
5 12.316793 7.373193 6.873905 10.242153 6.733338 11.53132 10.459047 12.321970
6  9.468645 7.578365 6.858501 10.613006 6.718292 11.48385  9.947725 11.479465
      AQP9    ASAH1      ATM    ATOX1    ATP7B      ATR    AURKA       BAD
1 6.896667 8.013413 8.991016 11.86723 10.99604 8.824009 9.089473 10.026779
2 7.300790 9.384410 9.218706 11.00517 10.42616 8.869149 9.643404  9.368393
3 6.929544 8.434568 8.650290 11.14605 10.19583 9.292160 9.564038  9.157618
4 6.754185 7.994920 9.151488 10.36164 11.95049 9.364890 9.245179  9.277187
5 7.190891 8.338853 8.413892 10.79890 11.53551 8.646506 9.464309 10.649963
6 7.029738 7.757668 9.661314 10.79775 10.76656 9.455848 9.622132  9.286238
      BAG1      BAG3      BAX     BCL2   BCL2A1   BCL2L1      BCR      BID
1 8.247202  8.196225 8.285454 7.061771 6.749110 11.52845 7.478098 8.776066
2 8.696358  8.902348 8.527747 6.923657 6.893357 11.52132 8.600347 8.610633
3 9.137049 10.419071 9.042224 6.782296 6.982602 10.94320 7.660845 8.412206
4 9.052640  9.072504 8.199018 6.960116 6.872465 11.60881 8.420708 7.888162
5 8.499520 10.096657 9.192863 7.054737 6.924474 11.37976 9.646834 9.267554
6 8.357150 10.060701 8.185528 6.969883 6.895864 12.05709 7.877872 8.481050
      BIRC2    BIRC3    BIRC5    BRCA1    BRCA2    CASP3    CASP8     CASP9
1  9.588253 9.120734 8.104391 8.229637 7.606711 8.156865 8.899888 11.856829
2  9.908033 9.838902 7.612666 8.216154 8.249681 7.886652 9.397077 10.377508
3  8.713247 6.861837 7.233072 7.599784 7.478098 8.722548 8.633693  9.513663
4  9.252486 7.334627 8.222438 7.773172 7.825928 7.643726 9.040616 11.105953
5  9.871594 8.805367 6.848384 8.257817 7.295718 8.485559 9.228429 11.620631
6 10.172727 7.896412 7.633207 7.800559 8.152128 7.938922 9.057885  9.975805
      CCL2     CCND1     CCNE1     CCNH     CCT8     CDC40     CDH1      CDK2
1 8.485123 11.237506  7.789426 7.731906 10.66772  9.406706 6.781030  8.861773
2 9.227227  9.940672  8.954642 8.516610 11.26844 10.115095 6.794842  9.412149
3 9.323911 12.750957  8.534080 8.614546 10.77727  9.068780 6.984990 10.132387
4 7.998110 11.441090  8.038295 8.325207 11.28383  8.745830 6.838876  9.477643
5 9.119603 13.649187  9.980143 8.056253 11.00517  9.106713 6.880984  7.992254
6 9.852250 12.280211 10.960413 8.103981 11.21451  9.455284 6.841160  9.530001
      CDK4   CDKN1A   CDKN1B    CDKN2A     CFL1    CFLAR     CFTR    CHEK1
1 10.98201 11.42440 9.383292  7.320087 12.47236 12.02346 6.684829 7.048884
2 10.74663 11.07151 8.996853  8.965756 13.08190 12.51807 6.822915 6.693706
3 11.81952 12.13184 8.620351  7.790862 12.68359 11.99563 6.693981 6.751534
4 10.64447 12.79570 8.049906  7.557507 11.58685 13.04655 6.704771 6.682156
5 10.17820 11.98798 9.213098  6.933126 12.35253 11.90179 6.719023 6.860076
6 10.21128 10.97491 9.271118 10.297463 12.31679 11.90704 6.740824 6.657458
   CIAPIN1     CLDN1    CLDN16    CLDN5     CLDN7      CLU   COX7A2   CTNNB1
1 7.681508  8.708947  7.845268 9.297974  8.338058 7.100871 12.08316 11.56230
2 7.534961  7.926664  7.281107 8.984927  9.550085 6.669287 12.33584 10.68571
3 8.174967 10.655983 11.076369 7.789044  8.605269 7.338735 11.77432 10.93122
4 7.324561  7.969987 10.187592 9.341966 10.588528 7.554864 12.35253 11.40614
5 7.367011  9.324963 10.416467 9.805089  8.059932 7.586933 10.81128 10.55741
6 7.943628 10.406669 12.253335 9.032984  9.119125 6.682610 12.14235 11.60459
    CYP2B6    CYP2E1   CYP3A4   CYP3A5    DAPK1   DIABLO     E2F1    ERBB2
1 7.226860 11.208238 7.734879 6.759182 7.743586 7.424097 7.553361 7.041597
2 7.101800  8.818492 6.606713 7.273960 8.114309 7.736911 9.366599 7.205736
3 6.704883  9.964152 6.633858 6.770068 7.653972 7.709740 7.949149 7.114339
4 6.689213 10.409367 7.027562 7.035601 8.003611 7.678017 7.629158 7.266241
5 6.695742 11.135486 7.115536 6.920464 7.242014 7.347626 8.484248 7.181896
6 6.723709  8.753469 6.602698 6.723458 7.093027 7.551000 9.007671 7.527508
     ERCC1    ERCC2    ERCC3    ERCC5     ERCC8     ETS1       F3     FADD
1 8.119690 6.634948 7.798293 8.375592  9.138738 9.317442 6.497447 8.554351
2 8.512343 8.204595 8.122248 9.597599  9.309278 9.440621 6.744717 8.687447
3 7.454531 7.234607 8.413416 8.782179  9.783796 8.162965 6.800483 9.209239
4 8.102303 7.108459 7.795757 8.744821  9.246290 8.114309 6.658022 8.613511
5 8.014282 6.605423 7.433361 8.567225 10.437541 8.871729 6.836299 9.524010
6 8.224209 7.625147 7.892648 8.870164  9.678867 7.993830 6.712306 8.824009
       FAS    FASLG      FN1       FOS     GART     GJA1     GLO1     GPX2
1 6.661529 6.964673 13.84285 11.834893  9.58040 9.958993 8.524575 6.909208
2 6.940935 7.224190 13.12844  9.774165 10.96740 8.066872 9.729713 6.795481
3 6.856376 6.805081 11.58972 11.241464 10.35577 7.872872 9.562123 6.698990
4 6.769281 6.722381 10.45366  8.293316 11.43317 8.384023 8.875251 6.733681
5 6.778335 7.051708 14.24558 11.396325 10.39101 9.211440 8.911571 6.589032
6 6.959689 7.004315 13.98307 10.381797 10.73191 8.489960 9.608504 6.669961
      GPX4    GSK3B       GSR      GSS    GSTA4    GSTK1     GSTM3     GSTM4
1 6.813452 12.15442  8.398504 8.961541 6.943554 10.32362 10.100971  9.568280
2 7.868558 12.43114 11.220488 7.614796 6.670250 10.43400  8.403353 10.132387
3 7.352592 12.69534  9.704405 7.509262 7.089369 10.72160  9.037159  9.091691
4 7.030146 12.76982  9.733084 8.431504 6.984135 10.82619  8.544503  9.750934
5 7.345952 12.18104  9.412149 8.516610 6.981129 10.28949  9.088984  9.044935
6 7.666169 12.71081 10.075754 8.083219 7.002652 10.01606  9.093766  9.511245
      GSTM5    GSTO1    GSTP1    GSTT1    GSTZ1      HAGH     HIF1A     HSPH1
1 10.306293 8.606291 11.78586 8.416559 7.291539  9.289830  9.491030  9.783796
2  7.872448 9.581561 12.37797 8.778741 7.521389  8.889739 10.110454 10.282395
3  6.983758 7.819526 11.39633 9.023617 7.188044  9.795539 10.382583 10.087814
4  8.406175 8.135362 11.26994 9.001613 7.692207 10.280731  9.825716  9.560779
5 10.084094 9.031920 12.57920 8.777632 7.675928  9.376039  9.642701  9.141591
6  7.612992 8.526893 11.79920 9.071992 7.034782  9.344247 10.097440  9.936790
     IGF1R      IL6     INSR    ITGAE    KCNMA1      KIT     KLF1       LBR
1 6.838876 9.086239 6.800426 7.491948 10.257129 8.975076 6.889729  9.613636
2 6.875127 7.356368 6.702848 7.887034 10.926863 7.326361 6.814125 10.578780
3 7.079108 8.579867 6.759050 8.322547 10.579718 7.096939 6.974354 10.835462
4 6.957563 7.241782 6.629429 8.319912  9.944868 8.286431 6.864716  9.486122
5 6.942007 6.601503 6.822207 6.708006 10.454517 8.897156 6.877372  7.866780
6 7.103209 7.701416 6.772228 7.881255 10.433997 6.945788 6.852500 10.998302
      LIG4   MAP2K1     MAPK3    MAPK8     MCL1     MDM2      MGMT    MKI67
1 7.212709 8.125232 12.160799 6.760280 6.823211 8.373721  8.816099 12.49164
2 7.592506 8.994710 11.779350 6.830487 6.947933 8.272301 11.581831 11.50976
3 7.749951 8.815611 10.489889 6.821279 6.878306 8.728776 10.545614 10.84681
4 8.060331 7.805088 11.525632 6.725328 6.745341 8.798323 10.435745 12.43114
5 7.500034 7.733790 11.825978 6.955973 6.808960 7.974306 11.326666 12.91702
6 8.157733 8.143069  9.615703 6.871519 6.804433 7.806968 11.261863 12.13857
      MLH1     MLH3     MMP2      MMP9    MNAT1   MRE11A     MSH2     MSH3
1 9.572425 9.998742 6.702731 12.978979 7.803568 8.340568 6.980159 8.336425
2 9.386016 7.746810 7.630609 13.738296 7.877496 8.130434 6.932767 8.808049
3 8.983380 8.972458 7.756945  8.245050 8.770694 8.189652 7.287554 9.067189
4 9.197284 9.209753 7.144451  8.381403 9.090058 8.187218 7.071250 8.776066
5 9.587072 9.540095 9.694220 12.311452 8.386363 7.600436 7.044314 7.894203
6 8.377317 8.089658 8.114692 11.661257 8.863229 8.192236 7.272901 8.648388
      MSH6    MUTYH       MVP       MYC    NFKB1   NFKBIA    NR1H3    NR1H4
1 11.48785 8.441406 10.160548 10.863801 7.806222 14.01653 12.28710 6.753944
2 10.43314 8.087351 10.192472 12.136104 8.253187 14.30214 12.88488 6.766869
3 12.30680 8.106060  9.545129  9.928011 8.214850 14.30214 11.06684 6.848135
4 11.06815 8.009482 10.208878 10.804036 7.478761 13.64919 12.25334 6.778560
5 10.48989 8.059121 10.472425 10.493515 7.868558 14.02920 11.92540 6.942543
6 11.07922 7.855445 10.219166 10.781571 7.812260 13.91213 11.68122 6.824061
     NR1I3    NTRK2    PARP2    PDGFRB     PDK1   PIK3CA      POLB     POLI
1 7.164879 6.841160 8.358902  9.370726 6.903371 9.108833 11.120641 7.539652
2 6.812951 7.271305 7.477164  9.034046 7.127309 9.201932  9.964905 8.288179
3 6.798531 8.358488 8.145587 10.246274 7.256301 9.506519 10.098987 8.123014
4 6.888614 7.348783 9.838368  8.617009 6.903549 9.316871  9.659823 8.731196
5 7.389878 6.763306 7.256041  9.977210 6.993913 9.168068 10.835462 8.108127
6 6.742970 7.317526 7.720117  9.911881 7.013408 9.879251 10.386700 7.974730
     PSMA3      RAD1    RAD23A   RAD23B    RAD50    RAD51     RAF1      RB1
1 6.782166 13.030370  9.618876 8.843967 7.413681 7.440083 8.593359 8.702936
2 6.980542 10.503795 10.471470 9.246897 7.798293 6.702731 8.830427 8.617009
3 6.725328  8.564961  8.915122 9.076945 7.891816 6.926470 8.020952 9.674851
4 6.814259 12.015844 10.162910 9.877929 7.947961 7.935052 8.021755 8.412991
5 6.648600 12.268708  9.144911 8.536796 7.971156 8.088937 8.568614 7.822239
6 6.798531  8.672581 10.081694 8.153357 8.079344 6.611246 7.718002 8.936396
      RPL36    SGPP1    SIRT1    SIRT2    SIRT4    SIRT5  SLC10A1  SLC15A2
1 10.354919 6.917069 7.129892 7.144451 6.894916 10.97857 8.833532 7.010702
2 11.276217 7.143976 7.207968 6.662145 7.068340 10.12467 7.207738 7.099184
3 10.143224 7.205455 7.987801 6.566324 6.788607 10.98315 6.965629 7.292587
4  9.923729 6.753404 7.203537 6.982027 6.874808 11.09773 7.741941 7.080201
5 10.808039 7.116433 7.396476 6.953287 6.786749 10.89773 8.283368 7.082766
6 10.983148 6.921313 7.981465 7.454183 7.466898 11.31854 6.951303 7.343303
   SLC16A2  SLC16A3  SLC19A1  SLC19A2   SLC19A3   SLC1A4   SLC1A5  SLC22A1
1 7.935776 7.073882 8.302609 6.608728 10.319222 7.593181 6.722895 6.853897
2 7.306276 7.687788 8.035930 6.973975 11.249055 7.813252 6.843828 6.900822
3 8.088129 7.034543 7.397024 6.673266 10.179876 7.754429 6.794206 6.889729
4 7.656052 6.853179 8.308380 6.664070  8.808519 7.536985 6.837693 6.776230
5 7.268498 7.590332 9.109394 6.946268 12.026946 7.789426 6.767923 6.881709
6 7.061949 7.471210 8.479222 6.851816 10.216842 7.261172 6.847398 6.992752
   SLC28A1   SLC29A1   SLC29A2    SLC2A5  SLC31A1    SLC7A1  SLC7A10  SLC7A11
1 6.674104 10.014521  9.864982  9.048300 7.571797  8.920121 6.829303 7.247272
2 6.662544  9.981688 10.072713  9.798107 8.382291  9.969333 6.999004 6.633858
3 6.592974  9.491030  9.182233  7.686160 8.104815  9.705134 7.536985 7.488330
4 6.483127  9.713148  9.538248  7.253627 8.433297  8.508646 7.093263 7.495513
5 6.723320  9.668169  9.159178 10.168625 7.899521  9.433523 6.904507 6.656101
6 6.672712  9.431107  9.219328  8.476463 8.217021 10.246274 9.037159 7.903030
    SLC7A2    SLC7A5   SLC7A8 SLC9A3R2  SLCO4A1      SP1      SRC    STAT1
1 7.079565  9.537659 9.178516 9.574976 6.934997 8.724469 11.78092 14.12329
2 7.195008 11.765885 8.660329 8.738353 9.133377 8.947775 11.09672 13.52122
3 7.195264 10.746632 8.184732 8.696358 8.134120 9.789807 10.89362 12.96082
4 7.653623 11.041209 8.868709 9.726282 7.297339 8.790778 11.42155 13.23216
5 7.182167 10.971551 9.391325 8.800765 9.034046 7.951562 11.24649 13.50068
6 6.940789 10.242153 7.481978 9.236139 7.334928 8.708947 11.30645 13.22267
     STAT3   STAT5A   STAT5B     TGFA    TIMP1   TMEM109  TNFSF10     TOP1
1 10.38670 7.598387 7.370609 8.495975 12.94263  9.197852 7.350274 11.14493
2 11.44832 8.313083 8.451724 8.204998 13.42744  9.917066 7.985511 11.41880
3 10.81887 7.558533 8.701540 8.289002 14.44965 10.019861 6.702332 10.86906
4 10.79106 7.047179 7.844877 8.158572 13.50068  9.005625 6.777934 11.42440
5 10.72573 7.079323 7.921993 8.539952 14.22980  9.499840 7.336431 10.64622
6 11.00185 7.527831 8.021755 8.726781 13.11178  8.997311 8.475478 11.28510
      TOP2A     TOP2B      TP53   TP53BP2     TP73     TRADD    TRAF1      XPA
1  8.878214 10.313158  8.253187 10.893620 6.604516 10.627634 6.891011 6.557723
2 11.318544 10.403172  8.157733 10.405792 6.713857 11.563815 7.017953 6.634579
3 10.001815  9.987692  7.967948 10.090610 7.120558  9.583350 6.979454 6.644534
4 10.606993 10.428813 10.618854 10.079322 7.011689  9.879251 7.062134 6.713586
5 10.291002  9.870044  7.643020  9.946294 6.582531 11.636604 6.971541 6.680946
6 10.660817 10.509485  9.061171 11.044895 6.722895 10.745602 7.272121 6.692519
       XPC    XRCC1    XRCC2    XRCC3    XRCC4    XRCC5
1 11.46091 8.462342 6.820206 7.406896 6.749753 13.09007
2 10.57020 9.047179 7.347931 8.022978 7.009499 12.53423
3 10.79775 7.874741 7.123481 7.499060 6.685486 14.11107
4 11.44710 8.113415 6.991716 8.012213 6.775875 13.57871
5 11.33618 8.468848 6.863026 7.687788 6.880655 12.97898
6 10.24550 8.257817 7.140145 7.470006 6.844277 13.44681</code></pre>
</div>
</div>
</section>
<section id="contingency-tables-ct" class="slide level2">
<h2>Contingency tables (CT)</h2>
<p>CTs are tables that summarize 2 or more categorical variables using absolute or relative frequencies.</p>

<img data-src="img/contingency-table.svg" class="r-stretch quarto-figure-center"></section>
<section id="contingency-tables-ct-1" class="slide level2">
<h2>Contingency tables (CT)</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<table data-quarto-postprocess="true" style="border-collapse:collapse; border:none;">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<tbody>
<tr class="odd">
<td rowspan="2" data-quarto-table-cell-role="th" style="text-align: center; border-top: double; font-style: italic; font-weight: normal; border-bottom: 1px solid;">x</td>
<td colspan="2" data-quarto-table-cell-role="th" style="text-align: center; border-top: double; font-style: italic; font-weight: normal;">y</td>
<td rowspan="2" data-quarto-table-cell-role="th" style="text-align: center; border-top: double; font-style: italic; font-weight: normal; font-weight: bolder; font-style: italic; border-bottom: 1px solid;">Total</td>
</tr>
<tr class="even">
<td style="text-align: center; border-bottom: 1px solid; padding: 0.2cm;">1</td>
<td style="text-align: center; border-bottom: 1px solid; padding: 0.2cm;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding: 0.2cm; vertical-align: middle;">1</td>
<td style="text-align: center; padding: 0.2cm;"><span style="color:black;">43</span><br>
<span style="color:#993333;">28.7&nbsp;%</span></td>
<td style="text-align: center; padding: 0.2cm;"><span style="color:black;">38</span><br>
<span style="color:#993333;">25.3&nbsp;%</span></td>
<td style="text-align: center; padding: 0.2cm;"><span style="color:black;">81</span><br>
<span style="color:#993333;">54&nbsp;%</span></td>
</tr>
<tr class="even">
<td style="text-align: left; padding: 0.2cm; vertical-align: middle;">0</td>
<td style="text-align: center; padding: 0.2cm;"><span style="color:black;">37</span><br>
<span style="color:#993333;">24.7&nbsp;%</span></td>
<td style="text-align: center; padding: 0.2cm;"><span style="color:black;">32</span><br>
<span style="color:#993333;">21.3&nbsp;%</span></td>
<td style="text-align: center; padding: 0.2cm;"><span style="color:black;">69</span><br>
<span style="color:#993333;">46&nbsp;%</span></td>
</tr>
<tr class="odd">
<td style="text-align: left; padding: 0.2cm; border-bottom: double; font-weight: bolder; font-style: italic; vertical-align: middle;">Total</td>
<td style="text-align: center; padding: 0.2cm; border-bottom: double;"><span style="color:black;">80</span><br>
<span style="color:#993333;">53.3&nbsp;%</span></td>
<td style="text-align: center; padding: 0.2cm; border-bottom: double;"><span style="color:black;">70</span><br>
<span style="color:#993333;">46.7&nbsp;%</span></td>
<td style="text-align: center; padding: 0.2cm; border-bottom: double;"><span style="color:black;">150</span><br>
<span style="color:#993333;">100&nbsp;%</span></td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="contingency-tables-ct-2" class="slide level2">
<h2>Contingency tables (CT)</h2>
<p>Regardless of the content, contingency tables can be expressed with a common nomenclature.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<table class="table" data-quarto-postprocess="true" style="margin-left: auto; margin-right: auto;">
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th colspan="2" data-quarto-table-cell-role="th" style="text-align: center; empty-cells: hide; border-bottom: hidden;"></th>
<th colspan="2" data-quarto-table-cell-role="th" style="text-align: center; border-bottom: hidden; padding-bottom: 0; padding-left: 3px; padding-right: 3px;"><div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Truth
</div></th>
</tr>
<tr class="odd">
<th colspan="2" data-quarto-table-cell-role="th" style="text-align: center; empty-cells: hide; border-bottom: hidden;"></th>
<th data-quarto-table-cell-role="th" style="text-align: center; border-bottom: hidden; padding-bottom: 0; padding-left: 3px; padding-right: 3px;"><div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
1
</div></th>
<th data-quarto-table-cell-role="th" style="text-align: center; border-bottom: hidden; padding-bottom: 0; padding-left: 3px; padding-right: 3px;"><div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
0
</div></th>
</tr>
<tr class="header">
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td rowspan="2" style="text-align: center; font-weight: bold; vertical-align: middle !important;">Prediction</td>
<td style="text-align: center; font-weight: bold;">1</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">True Positive (TP)</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">False Positive (FP)</span></td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">0</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">False Negative (FN)</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">True Negative (TN)</span></td>
</tr>
</tbody>
</table>


</div>
</div>
</section>
<section id="contingency-tables-ct-3" class="slide level2">
<h2>Contingency tables (CT)</h2>
<p>There are a lot of metrics that can be calculated from a simple contingency tables:</p>
<div class="cell" data-layout-align="center">
<iframe src="https://en.wikipedia.org/wiki/Confusion_matrix" width="960" height="400px" data-external="1">
</iframe>
</div>
</section>
<section id="contingency-tables-ct-metrics" class="slide level2">
<h2>Contingency tables (CT) metrics</h2>
<p>The most important (and used) measures are:</p>
<ul>
<li>the <strong>Sensitivity</strong> (aka <strong>True Positive Rate</strong> TPR or <em>recall</em>) is <span class="math inline">\(\frac{TP}{(TP + FN)}\)</span></li>
<li>the <strong>Specificity</strong> (aka <strong>True Negative Rate</strong> TNR) is <span class="math inline">\(\frac{TN}{(FP + TN)}\)</span></li>
<li>the <strong>Accuracy</strong> is <span class="math inline">\(\frac{(TP + TN)}{(TP + TN + FP + FN)}\)</span></li>
<li>the <strong>Positive Predictive Value</strong> is <span class="math inline">\(\frac{TP}{TP + FP}\)</span> or <span class="math inline">\(\frac{TPR \times \rho}{TPR \times \rho + (1 - TPR) \times (1 - \rho)}\)</span> where <span class="math inline">\(\rho\)</span> is the prevalence i.e.&nbsp;<span class="math inline">\(TP + FP\)</span></li>
<li>Area Under the Curve (AUC) is the area under the ROC curve that represent classification performance</li>
</ul>
</section>
<section id="contingency-tables-ct-metrics-1" class="slide level2">
<h2>Contingency tables (CT) metrics</h2>
<div class="cell" data-layout-align="center">

<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="2">
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Truth
</div>
</th>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="1">
</th>
</tr>
<tr>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="2">
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
1
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
0
</div>
</th>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="1">
</th>
</tr>
<tr>
<th style="text-align:center;">
</th>
<th style="text-align:center;">
</th>
<th style="text-align:center;">
</th>
<th style="text-align:center;">
</th>
<th style="text-align:center;">
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;font-weight: bold;vertical-align: middle !important;" rowspan="2">
Test
</td>
<td style="text-align:center;font-weight: bold;">
1
</td>
<td style="text-align:center;">
<span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">True Positive (TP)</span>
</td>
<td style="text-align:center;">
<span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">False Positive (FP)</span>
</td>
<td style="text-align:center;">
<span class="math inline">\(PPV = \frac{TP}{TP + FP}\)</span>
</td>
</tr>
<tr>
<td style="text-align:center;font-weight: bold;">
0
</td>
<td style="text-align:center;">
<span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">False Negative (FN)</span>
</td>
<td style="text-align:center;">
<span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">True Negative (TN)</span>
</td>
<td style="text-align:center;">
<span class="math inline">\(NPV = \frac{TN}{TN + FN}\)</span>
</td>
</tr>
<tr>
<td style="text-align:center;font-weight: bold;vertical-align: middle !important;" rowspan="2">
</td>
<td style="text-align:center;font-weight: bold;">
</td>
<td style="text-align:center;">
<span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;"><span class="math inline">\(Sensitivity = \frac{TP}{TP + FN}\)</span></span>
</td>
<td style="text-align:center;">
<span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;"><span class="math inline">\(Specificity = \frac{TN}{TN + FP}\)</span></span>
</td>
<td style="text-align:center;vertical-align: middle !important;" rowspan="2">
<span class="math inline">\(N = TP + FP + FN + TN\)</span>
</td>
</tr>
<tr>
<td style="text-align:center;font-weight: bold;">
</td>
<td style="text-align:center;">
<span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;"><span class="math inline">\(\rho = \frac{TP + FN}{N}\)</span></span>
</td>
<td style="text-align:center;">
<span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;"><span class="math inline">\(1 - \rho = \frac{FP + TN}{N}\)</span></span>
</td>
</tr>
</tbody>

</table>
</div>
</section>
<section id="contingency-tables-ct-metrics-2" class="slide level2">
<h2>Contingency tables (CT) metrics</h2>
<div class="cell" data-layout-align="center">

<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="2">
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="2">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Truth
</div>
</th>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="1">
</th>
</tr>
<tr>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="2">
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
1
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
0
</div>
</th>
<th style="empty-cells: hide;border-bottom:hidden;" colspan="1">
</th>
</tr>
<tr>
<th style="text-align:center;">
</th>
<th style="text-align:center;">
</th>
<th style="text-align:center;">
</th>
<th style="text-align:center;">
</th>
<th style="text-align:center;">
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;font-weight: bold;vertical-align: middle !important;" rowspan="2">
Test
</td>
<td style="text-align:center;font-weight: bold;">
1
</td>
<td style="text-align:center;">
<span class="math inline">\(Sensitivity \times \rho\)</span>
</td>
<td style="text-align:center;">
<span class="math inline">\((1 - Specificity) \times (1 - \rho)\)</span>
</td>
<td style="text-align:center;">
<span class="math inline">\(PPV = \frac{Sensitivity \times \rho}{Sensitivity \times \rho + (1 - Sensitivity) \times (1 - \rho)}\)</span>
</td>
</tr>
<tr>
<td style="text-align:center;font-weight: bold;">
0
</td>
<td style="text-align:center;">
<span class="math inline">\((1 - Sensitivity) \times \rho\)</span>
</td>
<td style="text-align:center;">
<span class="math inline">\(Specificity \times (1 - \rho)\)</span>
</td>
<td style="text-align:center;">
<span class="math inline">\(NPV = \frac{Specificity \times (1 - \rho)}{(1 - Sensitivity) \times \rho + Specificity \times (1 - \rho)}\)</span>
</td>
</tr>
<tr>
<td style="text-align:center;font-weight: bold;vertical-align: middle !important;" rowspan="2">
</td>
<td style="text-align:center;font-weight: bold;">
</td>
<td style="text-align:center;">
<span class="math inline">\(Sensitivity = \frac{TP}{TP + FN}\)</span>
</td>
<td style="text-align:center;">
<span class="math inline">\(Specificity = \frac{TN}{TN + FP}\)</span>
</td>
<td style="text-align:center;vertical-align: middle !important;" rowspan="2">
1
</td>
</tr>
<tr>
<td style="text-align:center;font-weight: bold;">
</td>
<td style="text-align:center;">
<span class="math inline">\(\rho\)</span>
</td>
<td style="text-align:center;">
<span class="math inline">\(1 - \rho\)</span>
</td>
</tr>
</tbody>

</table>
</div>
</section>
<section id="sensitivity-and-specificity" class="slide level2">
<h2>Sensitivity and Specificity</h2>
<p>To better understand the <em>sensitivity</em> and <em>specificity</em> we can use several formulations. <em>Sensitivity</em> and <em>Specificity</em> (and also other metrics) are essentially conditional probabilities.</p>
<p><span class="math display">\[
Sensitivity = p(T^+|S^+) = \frac{p(S^+|T^+)p(T^+)}{p(S^+)} = \frac{\frac{TP}{TP + FP}\frac{TP + FP}{N}}{\frac{TP + FN}{N}} = \frac{PPVp(T^+)}{\rho}
\]</span></p>
<p><span class="math display">\[
Specificity = p(T^-|S^-) = \frac{p(S^-|T^-)p(T^-)}{p(S^-)} = \frac{\frac{TN}{TN + FN}\frac{TN + FN}{N}}{\frac{FP + TN}{N}} = \frac{NPVp(T^-)}{1 -\rho}
\]</span></p>
</section>
<section id="positive-predictive-value-ppv" class="slide level2">
<h2>Positive Predictive Value (PPV)</h2>
<p>The PPV is the probability of having a diseases given that my test is positive. While sensitivity and specificity are generally stable regardless the prevalence <span class="citation" data-cites="Brenner1997-yq">(but see <a href="#/references" role="doc-biblioref" onclick="">Brenner and Gefeller 1997</a>)</span>, PPV is strongly affected by the disease prevalence.</p>
<p>Let‚Äôs write a function to calculate the PPV:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>ppv <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">tp =</span> <span class="cn">NULL</span>,</span>
<span id="cb14-2"><a href="#cb14-2"></a>                <span class="at">fp =</span> <span class="cn">NULL</span>, </span>
<span id="cb14-3"><a href="#cb14-3"></a>                <span class="at">sensitivity =</span> <span class="cn">NULL</span>, </span>
<span id="cb14-4"><a href="#cb14-4"></a>                <span class="at">specificity =</span> <span class="cn">NULL</span>, </span>
<span id="cb14-5"><a href="#cb14-5"></a>                <span class="at">prevalence =</span> <span class="cn">NULL</span>){</span>
<span id="cb14-6"><a href="#cb14-6"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(tp) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(fp)){</span>
<span id="cb14-7"><a href="#cb14-7"></a>    tp <span class="sc">/</span> (tp <span class="sc">+</span> fp)</span>
<span id="cb14-8"><a href="#cb14-8"></a>  } <span class="cf">else</span> <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(sensitivity) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(specificity) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(prevalence)){</span>
<span id="cb14-9"><a href="#cb14-9"></a>    (sensitivity <span class="sc">*</span> prevalence) <span class="sc">/</span> (sensitivity <span class="sc">*</span> prevalence <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> specificity) <span class="sc">*</span> (<span class="dv">1</span> <span class="sc">-</span> prevalence))</span>
<span id="cb14-10"><a href="#cb14-10"></a>  } <span class="cf">else</span>{</span>
<span id="cb14-11"><a href="#cb14-11"></a>    <span class="fu">stop</span>(<span class="st">"To calculate PPV tp AND fp OR sensitivity AND specificity AND prevalence need to be specified!"</span>)</span>
<span id="cb14-12"><a href="#cb14-12"></a>  }</span>
<span id="cb14-13"><a href="#cb14-13"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="positive-predictive-value-ppv-1" class="slide level2">
<h2>Positive Predictive Value (PPV)</h2>
<p>Now we can calculate the PPV, fixing <em>sensitivity</em> and <em>specificity</em> by changing the <strong>prevalence</strong>:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="co"># let's vary the prevalence</span></span>
<span id="cb15-2"><a href="#cb15-2"></a>prevalence <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.01</span>)</span>
<span id="cb15-3"><a href="#cb15-3"></a></span>
<span id="cb15-4"><a href="#cb15-4"></a><span class="co"># computing ppv for each prevalence, fixing the specificity and sensitivity</span></span>
<span id="cb15-5"><a href="#cb15-5"></a>ppvs <span class="ot">&lt;-</span> <span class="fu">ppv</span>(<span class="at">sensitivity =</span> <span class="fl">0.9</span>, <span class="at">specificity =</span> <span class="fl">0.8</span>, <span class="at">prevalence =</span> prevalence)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-24-1.svg" width="960" class="r-stretch quarto-figure-center"></section>
<section id="youdens-j-disclaimer" class="slide level2">
<h2>Youden‚Äôs J, disclaimer</h2>
<p>The Youden‚Äôs J is just an option among several alternatives for choosing the best threshold. For example the <code>cutpointr</code> package:</p>
<div class="cell" data-layout-align="center">
<iframe src="https://cran.r-project.org/web/packages/cutpointr/vignettes/cutpointr.html#:~:text=in%20both%20classes-,Metric%20functions,-The%20included%20metrics" width="960" height="400px" data-external="1">
</iframe>
</div>
</section>
<section id="an-example" class="slide level2">
<h2>An example‚Ä¶</h2>
<p>Let‚Äôs imagine to have a group of patients with a certain disease (called <span class="math inline">\(S\)</span>) or condition measured by a <strong>gold-standard</strong> test. We are developing a new faster and cheaper test (called <span class="math inline">\(T\)</span>) but we are not sure about the capacity to reliably detect the condition.</p>
<div class="cell" data-layout-align="center">
<details>
<summary>Simulation code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>dat_ex <span class="ot">&lt;-</span> <span class="fu">sim_bin_class</span>(<span class="fl">0.75</span>, <span class="at">n =</span> <span class="dv">100</span>, <span class="at">prevalence =</span> <span class="fl">0.2</span>, <span class="at">var.names =</span> <span class="fu">c</span>(<span class="at">x =</span> <span class="st">"test"</span>, <span class="at">y =</span> <span class="st">"state"</span>))</span>
<span id="cb16-2"><a href="#cb16-2"></a>dat_ex<span class="sc">$</span>state <span class="ot">&lt;-</span> <span class="fu">factor</span>(dat_ex<span class="sc">$</span>state, <span class="at">levels =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"sick"</span>, <span class="st">"healthy"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code>    state   test
1 healthy -1.065
2 healthy -1.138
3 healthy  -1.46
4 healthy  0.811
5     ...    ...
6    sick  1.705
7    sick -0.305
8    sick  1.075
9    sick  0.913</code></pre>
</div>
</div>
</section>
<section id="binary-classifiers" class="slide level2">
<h2>Binary classifiers</h2>
<div class="panel-tabset">
<ul id="tabset-1" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-1-1">Summary</a></li><li><a href="#tabset-1-2">Dotplot</a></li><li><a href="#tabset-1-3">Density Plot</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1">
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co"># total</span></span>
<span id="cb18-2"><a href="#cb18-2"></a><span class="fu">nrow</span>(dat_ex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 100</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="co"># prevalence</span></span>
<span id="cb20-2"><a href="#cb20-2"></a><span class="fu">table</span>(dat_ex<span class="sc">$</span>state)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   sick healthy 
     20      80 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="fu">table</span>(dat_ex<span class="sc">$</span>state) <span class="sc">/</span> <span class="fu">nrow</span>(dat_ex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   sick healthy 
    0.2     0.8 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="co"># difference in the test between the two groups</span></span>
<span id="cb24-2"><a href="#cb24-2"></a><span class="fu">tapply</span>(dat_ex<span class="sc">$</span>test, dat_ex<span class="sc">$</span>state, mean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      sick    healthy 
 0.4308099 -0.3869299 </code></pre>
</div>
</div>
</div>
<div id="tabset-1-2">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-29-1.svg" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-1-3">
<p>Probably the most intuitive way is the density plot for each group:</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-30-1.svg" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="binary-classifiers-1" class="slide level2">
<h2>Binary classifiers</h2>
<p>Intuitively, as the mean difference (on the predictor) between the two groups increase the two groups are easy to discriminate:</p>

<img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-31-1.svg" width="960" class="r-stretch quarto-figure-center"></section>
<section id="binary-classifiers-thresholds" class="slide level2">
<h2>Binary classifiers, Thresholds</h2>
<p>To create a contingency table and calculate the related metrics we need to choose a threshold on the predictor variable:</p>
<div class="panel-tabset">
<ul id="tabset-2" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-2-1">Threshold = 0</a></li><li><a href="#tabset-2-2">Threshold = -1.5</a></li><li><a href="#tabset-2-3">Threshold = 2</a></li></ul>
<div class="tab-content">
<div id="tabset-2-1">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-33-1.svg" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-2-2">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-34-1.svg" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-2-3">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-35-1.svg" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="binary-classifiers-thresholds-1" class="slide level2">
<h2>Binary classifiers, Thresholds</h2>
<p>Similarly, we can create the contingency tables with related metrics:</p>
<div class="panel-tabset">
<ul id="tabset-3" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-3-1">Threshold = 0</a></li><li><a href="#tabset-3-2">Threshold = -1.5</a></li><li><a href="#tabset-3-3">Threshold = 2</a></li></ul>
<div class="tab-content">
<div id="tabset-3-1">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<table class="table" data-quarto-postprocess="true" style="font-size: 20px; margin-left: auto; margin-right: auto;border-bottom: 0;">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th colspan="2" data-quarto-table-cell-role="th" style="text-align: center; empty-cells: hide; border-bottom: hidden;"></th>
<th colspan="3" data-quarto-table-cell-role="th" style="text-align: center; border-bottom: hidden; padding-bottom: 0; padding-left: 3px; padding-right: 3px;"><div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Truth
</div></th>
</tr>
<tr class="odd">
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th">1</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">0</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">Tot</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td rowspan="3" style="text-align: center; font-weight: bold; vertical-align: middle !important;">Test</td>
<td style="text-align: center; font-weight: bold;">1</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">4035</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">937</span></td>
<td style="text-align: center;">4972</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">0</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">965</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">4063</span></td>
<td style="text-align: center;">5028</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">Tot</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;">5000</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;">5000</span></td>
<td style="text-align: center;">10000</td>
</tr>
</tbody><tfoot>
<tr class="odd">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>Threshold</strong> = 0</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>Sensitivity</strong> = 0.807</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>Specificity</strong> = 0.813</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>PPV</strong> = 0.812</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
</tfoot>

</table>


</div>
</div>
</div>
<div id="tabset-3-2">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<table class="table" data-quarto-postprocess="true" style="font-size: 20px; margin-left: auto; margin-right: auto;border-bottom: 0;">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th colspan="2" data-quarto-table-cell-role="th" style="text-align: center; empty-cells: hide; border-bottom: hidden;"></th>
<th colspan="3" data-quarto-table-cell-role="th" style="text-align: center; border-bottom: hidden; padding-bottom: 0; padding-left: 3px; padding-right: 3px;"><div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Truth
</div></th>
</tr>
<tr class="odd">
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th">1</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">0</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">Tot</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td rowspan="3" style="text-align: center; font-weight: bold; vertical-align: middle !important;">Test</td>
<td style="text-align: center; font-weight: bold;">1</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">4958</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">3672</span></td>
<td style="text-align: center;">8630</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">0</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">42</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">1328</span></td>
<td style="text-align: center;">1370</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">Tot</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;">5000</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;">5000</span></td>
<td style="text-align: center;">10000</td>
</tr>
</tbody><tfoot>
<tr class="odd">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>Threshold</strong> = -1.5</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>Sensitivity</strong> = 0.992</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>Specificity</strong> = 0.266</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>PPV</strong> = 0.575</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
</tfoot>

</table>


</div>
</div>
</div>
<div id="tabset-3-3">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<table class="table" data-quarto-postprocess="true" style="font-size: 20px; margin-left: auto; margin-right: auto;border-bottom: 0;">
<colgroup>
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
<col style="width: 20%">
</colgroup>
<thead>
<tr class="header">
<th colspan="2" data-quarto-table-cell-role="th" style="text-align: center; empty-cells: hide; border-bottom: hidden;"></th>
<th colspan="3" data-quarto-table-cell-role="th" style="text-align: center; border-bottom: hidden; padding-bottom: 0; padding-left: 3px; padding-right: 3px;"><div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Truth
</div></th>
</tr>
<tr class="odd">
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th"></th>
<th style="text-align: center;" data-quarto-table-cell-role="th">1</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">0</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">Tot</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td rowspan="3" style="text-align: center; font-weight: bold; vertical-align: middle !important;">Test</td>
<td style="text-align: center; font-weight: bold;">1</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">653</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">6</span></td>
<td style="text-align: center;">659</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">0</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: #FC8D8D !important;">4347</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">4994</span></td>
<td style="text-align: center;">9341</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">Tot</td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;">5000</span></td>
<td style="text-align: center;"><span style="     border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: white !important;">5000</span></td>
<td style="text-align: center;">10000</td>
</tr>
</tbody><tfoot>
<tr class="odd">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>Threshold</strong> = 2</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>Sensitivity</strong> = 0.131</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>Specificity</strong> = 0.999</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center; padding: 0;"><sup></sup> <strong>PPV</strong> = 0.991</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
</tfoot>

</table>


</div>
</div>
</div>
</div>
</div>
</section>
<section id="roc" class="slide level2">
<h2>ROC</h2>
<p>We can start by plotting the specificity and sensitivity for a given threshold:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>cmat <span class="ot">&lt;-</span> <span class="fu">classify</span>(dat_ex_90, state, test, <span class="at">c =</span> <span class="dv">0</span>)</span>
<span id="cb26-2"><a href="#cb26-2"></a>cmat</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    tp  fp   tn  fn   tpr    fpr    tnr   fnr   tot prevalence c       ppv
1 4035 937 4063 965 0.807 0.1874 0.8126 0.193 10000        0.5 0 0.8115447
        npv
1 0.8080748</code></pre>
</div>
</div>
<div class="fragment">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-41-1.svg" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="roc-1" class="slide level2">
<h2>ROC</h2>
<p>When evaluating <strong>sensitivity</strong> and <strong>specificity</strong> with multiple thresholds we obtain the ROC curve:</p>
<div class="fragment">
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="img/roc-anim.gif" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="roc-2" class="slide level2">
<h2>ROC</h2>
<p>The Area Under the Curve (AUC) range between 0.5 (null classification) and 1 (perfect classification):</p>

<img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-43-1.svg" width="960" class="r-stretch quarto-figure-center"></section>
<section id="youdens-j" class="slide level2">
<h2>Youden‚Äôs J</h2>
<p>Given the trade-off between sensitivity and specificity, the choice of the best threshold (<span class="math inline">\(c\)</span>) is not straightforward. An option is using the Youden‚Äôs J:</p>
<p><span class="math display">\[
J = sensitivity + specificity - 1
\]</span> Thus taking the maximum of <span class="math inline">\(J_i\)</span> calculated on all threshold give us the threshold that maximize the sum between sensitivity and specificity.</p>
</section>
<section id="youdens-j-1" class="slide level2">
<h2>Youden‚Äôs J</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="fu">set.seed</span>(<span class="dv">104</span>)</span>
<span id="cb28-2"><a href="#cb28-2"></a>dat <span class="ot">&lt;-</span> <span class="fu">sim_bin_class</span>(<span class="at">auc =</span> <span class="fl">0.8</span>, <span class="at">n =</span> <span class="dv">100</span>, <span class="at">prevalence =</span> <span class="fl">0.5</span>)</span>
<span id="cb28-3"><a href="#cb28-3"></a>fit_roc <span class="ot">&lt;-</span> <span class="fu">roc</span>(y <span class="sc">~</span> x, <span class="at">data =</span> dat)</span>
<span id="cb28-4"><a href="#cb28-4"></a>J <span class="ot">&lt;-</span> <span class="fu">coords</span>(fit_roc, <span class="st">"best"</span>, <span class="at">best.method =</span> <span class="st">"youden"</span>)</span>
<span id="cb28-5"><a href="#cb28-5"></a>J<span class="sc">$</span>J <span class="ot">&lt;-</span> J<span class="sc">$</span>specificity <span class="sc">+</span> J<span class="sc">$</span>sensitivity <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb28-6"><a href="#cb28-6"></a>Js <span class="ot">&lt;-</span> fit_roc<span class="sc">$</span>sensitivities <span class="sc">+</span> fit_roc<span class="sc">$</span>specificities <span class="sc">-</span> <span class="dv">1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb29-2"><a href="#cb29-2"></a><span class="fu">plot</span>(fit_roc, <span class="at">main =</span> <span class="st">"ROC curve"</span>)</span>
<span id="cb29-3"><a href="#cb29-3"></a><span class="fu">points</span>(<span class="at">x =</span> J<span class="sc">$</span>specificity, <span class="at">y =</span> J<span class="sc">$</span>sensitivity, <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">col =</span> <span class="st">"firebrick"</span>, <span class="at">cex =</span> <span class="fl">1.4</span>)</span>
<span id="cb29-4"><a href="#cb29-4"></a></span>
<span id="cb29-5"><a href="#cb29-5"></a><span class="fu">plot</span>(fit_roc<span class="sc">$</span>thresholds, Js, <span class="at">type =</span> <span class="st">"l"</span>, <span class="at">xlab =</span> <span class="st">"Thresholds"</span>, <span class="at">ylab =</span> <span class="st">"Youden's J"</span>,</span>
<span id="cb29-6"><a href="#cb29-6"></a>     <span class="at">main =</span> <span class="st">"Youden's J"</span>)</span>
<span id="cb29-7"><a href="#cb29-7"></a><span class="fu">points</span>(J<span class="sc">$</span>threshold, <span class="fu">max</span>(Js), <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">col =</span> <span class="st">"firebrick"</span>, <span class="at">cex =</span> <span class="fl">1.4</span>)</span>
<span id="cb29-8"><a href="#cb29-8"></a><span class="fu">text</span>(<span class="at">x =</span> <span class="fl">0.6</span>, <span class="at">y =</span> <span class="fl">0.45</span>, <span class="at">labels =</span> <span class="fu">sprintf</span>(<span class="st">"Cutoff = %.2f"</span>, J<span class="sc">$</span>threshold))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-45-1.svg" width="960" class="r-stretch quarto-figure-center"></section>
<section id="data-simulation" class="slide level2 extra">
<h2>Data simulation</h2>
<p>All the example so far are based on simulated data. I wrote a little function (<code>sim_bin_class()</code>) that simulate a binary classifier assuming a latent probit model. You can play around to check what happens changing the parameters.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>sim_bin_class <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">auc =</span> <span class="cn">NULL</span>,</span>
<span id="cb30-2"><a href="#cb30-2"></a>                          <span class="at">d =</span> <span class="cn">NULL</span>,</span>
<span id="cb30-3"><a href="#cb30-3"></a>                          n, </span>
<span id="cb30-4"><a href="#cb30-4"></a>                          prevalence, </span>
<span id="cb30-5"><a href="#cb30-5"></a>                          <span class="at">x.mean =</span> <span class="dv">0</span>,</span>
<span id="cb30-6"><a href="#cb30-6"></a>                          <span class="at">x.sd =</span> <span class="dv">1</span>,</span>
<span id="cb30-7"><a href="#cb30-7"></a>                          <span class="at">var.names =</span> <span class="fu">c</span>(<span class="at">x =</span> <span class="st">"x"</span>, <span class="at">y =</span> <span class="st">"y"</span>)){</span>
<span id="cb30-8"><a href="#cb30-8"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(auc) <span class="sc">&amp;</span> <span class="fu">is.null</span>(d)){</span>
<span id="cb30-9"><a href="#cb30-9"></a>    <span class="fu">stop</span>(<span class="st">"AUC or d need to be specified"</span>)</span>
<span id="cb30-10"><a href="#cb30-10"></a>  }</span>
<span id="cb30-11"><a href="#cb30-11"></a>  </span>
<span id="cb30-12"><a href="#cb30-12"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(d)){</span>
<span id="cb30-13"><a href="#cb30-13"></a>    d <span class="ot">&lt;-</span> <span class="fu">auc_to_d</span>(auc)</span>
<span id="cb30-14"><a href="#cb30-14"></a>  }</span>
<span id="cb30-15"><a href="#cb30-15"></a>  </span>
<span id="cb30-16"><a href="#cb30-16"></a>  n1 <span class="ot">&lt;-</span> <span class="fu">ceiling</span>(n <span class="sc">*</span> prevalence)</span>
<span id="cb30-17"><a href="#cb30-17"></a>  n0 <span class="ot">&lt;-</span> n <span class="sc">-</span> n1</span>
<span id="cb30-18"><a href="#cb30-18"></a>  y <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(n0, n1))</span>
<span id="cb30-19"><a href="#cb30-19"></a>  x <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(y <span class="sc">==</span> <span class="dv">1</span>, <span class="fu">rnorm</span>(n1, (d<span class="sc">/</span><span class="dv">2</span>)<span class="sc">*</span>x.sd <span class="sc">+</span> x.mean, <span class="dv">1</span>), <span class="fu">rnorm</span>(n0, (<span class="sc">-</span>d<span class="sc">/</span><span class="dv">2</span>)<span class="sc">*</span>x.sd <span class="sc">+</span> x.mean, <span class="dv">1</span>))</span>
<span id="cb30-20"><a href="#cb30-20"></a>  dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(y, x)</span>
<span id="cb30-21"><a href="#cb30-21"></a>  <span class="fu">names</span>(dat) <span class="ot">&lt;-</span> var.names[<span class="fu">names</span>(dat)]</span>
<span id="cb30-22"><a href="#cb30-22"></a>  <span class="fu">return</span>(dat)</span>
<span id="cb30-23"><a href="#cb30-23"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-simulation-1" class="slide level2 extra">
<h2>Data simulation</h2>
<p>Let‚Äôs make an example simulating a binary classifier with an AUC of 0.7:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">sim_bin_class</span>(<span class="at">auc =</span> <span class="fl">0.7</span>, <span class="at">n =</span> <span class="dv">100</span>, <span class="at">prevalence =</span> <span class="fl">0.5</span>)</span>
<span id="cb31-2"><a href="#cb31-2"></a><span class="fu">head</span>(dat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  y          x
1 0  0.8100790
2 0 -0.4872855
3 0  0.2173173
4 0 -1.3149524
5 0 -0.5828904
6 0 -1.1558787</code></pre>
</div>
</div>
</section>
<section id="data-simulation-2" class="slide level2 extra">
<h2>Data simulation</h2>
<p>Let‚Äôs see some useful descriptive plots:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="fu">plot_class</span>(dat, y, x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-48-1.svg" width="960" class="r-stretch quarto-figure-center"></section>
<section id="data-simulation-3" class="slide level2 extra">
<h2>Data simulation</h2>
<p>The <code>plot_class()</code> function is a shortcut to produce the dotplot-density combination that is useful when plotting a binary classifier:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>plot_class <span class="ot">&lt;-</span> <span class="cf">function</span>(data, y, x){</span>
<span id="cb34-2"><a href="#cb34-2"></a>  </span>
<span id="cb34-3"><a href="#cb34-3"></a>  x <span class="ot">&lt;-</span> <span class="fu">deparse</span>(<span class="fu">substitute</span>(x))</span>
<span id="cb34-4"><a href="#cb34-4"></a>  y <span class="ot">&lt;-</span> <span class="fu">deparse</span>(<span class="fu">substitute</span>(y))</span>
<span id="cb34-5"><a href="#cb34-5"></a>  </span>
<span id="cb34-6"><a href="#cb34-6"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.numeric</span>(data[[y]])) <span class="fu">stop</span>(<span class="st">"The response variable need to be numeric (0-1)"</span>)</span>
<span id="cb34-7"><a href="#cb34-7"></a></span>
<span id="cb34-8"><a href="#cb34-8"></a>  dens <span class="ot">&lt;-</span> <span class="fu">tapply</span>(data[[x]], data[[y]], density)</span>
<span id="cb34-9"><a href="#cb34-9"></a>  ndens <span class="ot">&lt;-</span> <span class="fu">names</span>(dens)</span>
<span id="cb34-10"><a href="#cb34-10"></a>  </span>
<span id="cb34-11"><a href="#cb34-11"></a>  fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(y <span class="sc">~</span> x, <span class="at">data =</span> data, <span class="at">family =</span> <span class="fu">binomial</span>(<span class="at">link =</span> <span class="st">"logit"</span>))</span>
<span id="cb34-12"><a href="#cb34-12"></a>  def.par <span class="ot">=</span> <span class="fu">par</span>(<span class="at">no.readonly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb34-13"><a href="#cb34-13"></a>  </span>
<span id="cb34-14"><a href="#cb34-14"></a>  xlim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">min</span>(<span class="fu">sapply</span>(dens, <span class="cf">function</span>(x) <span class="fu">min</span>(x<span class="sc">$</span>x))),</span>
<span id="cb34-15"><a href="#cb34-15"></a>            <span class="fu">max</span>(<span class="fu">sapply</span>(dens, <span class="cf">function</span>(x) <span class="fu">max</span>(x<span class="sc">$</span>x))))</span>
<span id="cb34-16"><a href="#cb34-16"></a>  </span>
<span id="cb34-17"><a href="#cb34-17"></a>  ylim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fu">sapply</span>(dens, <span class="cf">function</span>(x) <span class="fu">max</span>(x<span class="sc">$</span>y))))</span>
<span id="cb34-18"><a href="#cb34-18"></a>  </span>
<span id="cb34-19"><a href="#cb34-19"></a>  <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb34-20"><a href="#cb34-20"></a>  </span>
<span id="cb34-21"><a href="#cb34-21"></a>  <span class="fu">plot</span>(dens[[<span class="dv">1</span>]],</span>
<span id="cb34-22"><a href="#cb34-22"></a>       <span class="at">main =</span> <span class="st">"Density Plot"</span>,</span>
<span id="cb34-23"><a href="#cb34-23"></a>       <span class="at">xlab =</span> x,</span>
<span id="cb34-24"><a href="#cb34-24"></a>       <span class="at">col =</span> <span class="st">"firebrick"</span>,</span>
<span id="cb34-25"><a href="#cb34-25"></a>       <span class="at">lwd =</span> <span class="fl">1.5</span>,</span>
<span id="cb34-26"><a href="#cb34-26"></a>       <span class="at">xlim =</span> xlim,</span>
<span id="cb34-27"><a href="#cb34-27"></a>       <span class="at">ylim =</span> ylim)</span>
<span id="cb34-28"><a href="#cb34-28"></a>  <span class="fu">rug</span>(data[[x]][data[[y]] <span class="sc">==</span> ndens[<span class="dv">1</span>]], <span class="at">col =</span> <span class="st">"firebrick"</span>, <span class="at">lwd =</span> <span class="fl">1.5</span>)</span>
<span id="cb34-29"><a href="#cb34-29"></a>  <span class="fu">rug</span>(data[[x]][data[[y]] <span class="sc">==</span> ndens[<span class="dv">2</span>]], <span class="at">col =</span> <span class="st">"dodgerblue3"</span>, <span class="at">lwd =</span> <span class="fl">1.5</span>)</span>
<span id="cb34-30"><a href="#cb34-30"></a>  <span class="fu">lines</span>(dens[[<span class="dv">2</span>]], <span class="at">col =</span> <span class="st">"dodgerblue3"</span>, <span class="at">lwd =</span> <span class="fl">1.5</span>)</span>
<span id="cb34-31"><a href="#cb34-31"></a>  </span>
<span id="cb34-32"><a href="#cb34-32"></a>  xr <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">min</span>(data[[x]]), <span class="fu">max</span>(data[[x]]), <span class="at">length.out =</span> <span class="dv">500</span>)</span>
<span id="cb34-33"><a href="#cb34-33"></a>  pdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(xr)</span>
<span id="cb34-34"><a href="#cb34-34"></a>  <span class="fu">names</span>(pdata) <span class="ot">&lt;-</span> x</span>
<span id="cb34-35"><a href="#cb34-35"></a>  pr <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit, <span class="at">newdata =</span> pdata, <span class="at">type =</span> <span class="st">"response"</span>)</span>
<span id="cb34-36"><a href="#cb34-36"></a>  </span>
<span id="cb34-37"><a href="#cb34-37"></a>  y_jitt <span class="ot">&lt;-</span> <span class="fu">jitter</span>(data[[y]], <span class="fl">0.2</span>)</span>
<span id="cb34-38"><a href="#cb34-38"></a>  y_col <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(data[[y]] <span class="sc">==</span> ndens[<span class="dv">1</span>], <span class="st">"firebrick"</span>, <span class="st">"dodgerblue3"</span>)</span>
<span id="cb34-39"><a href="#cb34-39"></a>  </span>
<span id="cb34-40"><a href="#cb34-40"></a>  <span class="fu">plot</span>(data[[x]], y_jitt, </span>
<span id="cb34-41"><a href="#cb34-41"></a>       <span class="at">col =</span> scales<span class="sc">::</span><span class="fu">alpha</span>(y_col, <span class="fl">0.4</span>),</span>
<span id="cb34-42"><a href="#cb34-42"></a>       <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb34-43"><a href="#cb34-43"></a>       <span class="at">main =</span> <span class="st">"Dotplot"</span>,</span>
<span id="cb34-44"><a href="#cb34-44"></a>       <span class="at">xlab =</span> x,</span>
<span id="cb34-45"><a href="#cb34-45"></a>       <span class="at">ylab =</span> y)</span>
<span id="cb34-46"><a href="#cb34-46"></a>  <span class="fu">lines</span>(pdata[[<span class="dv">1</span>]], pr)</span>
<span id="cb34-47"><a href="#cb34-47"></a>  </span>
<span id="cb34-48"><a href="#cb34-48"></a>  <span class="fu">par</span>(def.par)</span>
<span id="cb34-49"><a href="#cb34-49"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-simulation-4" class="slide level2 extra">
<h2>Data Simulation</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>fit_roc <span class="ot">&lt;-</span> pROC<span class="sc">::</span><span class="fu">roc</span>(y <span class="sc">~</span> x, <span class="at">data =</span> dat)</span>
<span id="cb35-2"><a href="#cb35-2"></a>fit_roc</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
roc.formula(formula = y ~ x, data = dat)

Data: x in 50 controls (y 0) &lt; 50 cases (y 1).
Area under the curve: 0.742</code></pre>
</div>
</div>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>fit_roc <span class="sc">|&gt;</span> </span>
<span id="cb37-2"><a href="#cb37-2"></a>  pROC<span class="sc">::</span><span class="fu">ggroc</span>() <span class="sc">+</span> </span>
<span id="cb37-3"><a href="#cb37-3"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"darkgrey"</span>) <span class="sc">+</span></span>
<span id="cb37-4"><a href="#cb37-4"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb37-5"><a href="#cb37-5"></a>  <span class="fu">theme_minimal</span>(<span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb37-6"><a href="#cb37-6"></a>  <span class="fu">xlab</span>(<span class="st">"Specificity"</span>) <span class="sc">+</span></span>
<span id="cb37-7"><a href="#cb37-7"></a>  <span class="fu">ylab</span>(<span class="st">"Sensitivity"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-51-1.svg" width="960" class="r-stretch quarto-figure-center"></section>
<section id="data-simulation-5" class="slide level2 extra">
<h2>Data Simulation</h2>
<p>Or manually using the <code>classify()</code> function that compute common classification metrics given one or more thresholds <span class="math inline">\(c_i\)</span>:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>classify <span class="ot">&lt;-</span> <span class="cf">function</span>(data, y, x, c, <span class="at">na.rm =</span> <span class="cn">FALSE</span>){</span>
<span id="cb38-2"><a href="#cb38-2"></a>  xn <span class="ot">&lt;-</span> <span class="fu">deparse</span>(<span class="fu">substitute</span>(x))</span>
<span id="cb38-3"><a href="#cb38-3"></a>  yn <span class="ot">&lt;-</span> <span class="fu">deparse</span>(<span class="fu">substitute</span>(y))</span>
<span id="cb38-4"><a href="#cb38-4"></a>  </span>
<span id="cb38-5"><a href="#cb38-5"></a>  <span class="cf">if</span>(na.rm){</span>
<span id="cb38-6"><a href="#cb38-6"></a>    data <span class="ot">&lt;-</span> data[<span class="fu">complete.cases</span>(data[, <span class="fu">c</span>(xn, yn)]), ]</span>
<span id="cb38-7"><a href="#cb38-7"></a>  }</span>
<span id="cb38-8"><a href="#cb38-8"></a>  </span>
<span id="cb38-9"><a href="#cb38-9"></a>  confusion <span class="ot">&lt;-</span> <span class="fu">lapply</span>(c, <span class="cf">function</span>(cr){</span>
<span id="cb38-10"><a href="#cb38-10"></a>    <span class="co"># classify based on c</span></span>
<span id="cb38-11"><a href="#cb38-11"></a>    yp <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(data[[xn]] <span class="sc">&gt;=</span> cr, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb38-12"><a href="#cb38-12"></a>    </span>
<span id="cb38-13"><a href="#cb38-13"></a>    out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb38-14"><a href="#cb38-14"></a>      <span class="at">tp =</span> <span class="fu">sum</span>(data[[yn]] <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> yp <span class="sc">==</span> <span class="dv">1</span>),</span>
<span id="cb38-15"><a href="#cb38-15"></a>      <span class="at">fp =</span> <span class="fu">sum</span>(data[[yn]] <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> yp <span class="sc">==</span> <span class="dv">1</span>),</span>
<span id="cb38-16"><a href="#cb38-16"></a>      <span class="at">tn =</span> <span class="fu">sum</span>(data[[yn]] <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> yp <span class="sc">==</span> <span class="dv">0</span>),</span>
<span id="cb38-17"><a href="#cb38-17"></a>      <span class="at">fn =</span> <span class="fu">sum</span>(data[[yn]] <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> yp <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb38-18"><a href="#cb38-18"></a>    )</span>
<span id="cb38-19"><a href="#cb38-19"></a>    </span>
<span id="cb38-20"><a href="#cb38-20"></a>    <span class="co"># rates</span></span>
<span id="cb38-21"><a href="#cb38-21"></a>    out<span class="sc">$</span>tpr <span class="ot">&lt;-</span> <span class="fu">with</span>(out, tp <span class="sc">/</span> (tp <span class="sc">+</span> fn))</span>
<span id="cb38-22"><a href="#cb38-22"></a>    out<span class="sc">$</span>fpr <span class="ot">&lt;-</span> <span class="fu">with</span>(out, fp <span class="sc">/</span> (fp <span class="sc">+</span> tn))</span>
<span id="cb38-23"><a href="#cb38-23"></a>    out<span class="sc">$</span>tnr <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> out<span class="sc">$</span>fpr</span>
<span id="cb38-24"><a href="#cb38-24"></a>    out<span class="sc">$</span>fnr <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> out<span class="sc">$</span>tpr</span>
<span id="cb38-25"><a href="#cb38-25"></a>    out<span class="sc">$</span>tot <span class="ot">&lt;-</span> <span class="fu">nrow</span>(data)</span>
<span id="cb38-26"><a href="#cb38-26"></a>    out<span class="sc">$</span>prevalence <span class="ot">&lt;-</span> <span class="fu">mean</span>(data[[yn]] <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb38-27"><a href="#cb38-27"></a>    out<span class="sc">$</span>c <span class="ot">&lt;-</span> cr</span>
<span id="cb38-28"><a href="#cb38-28"></a>    out<span class="sc">$</span>ppv <span class="ot">&lt;-</span> <span class="fu">with</span>(out, tp <span class="sc">/</span> (tp <span class="sc">+</span> fp))</span>
<span id="cb38-29"><a href="#cb38-29"></a>    out<span class="sc">$</span>npv <span class="ot">&lt;-</span> <span class="fu">with</span>(out, tn <span class="sc">/</span> (tn <span class="sc">+</span> fn))</span>
<span id="cb38-30"><a href="#cb38-30"></a>    out</span>
<span id="cb38-31"><a href="#cb38-31"></a>  })</span>
<span id="cb38-32"><a href="#cb38-32"></a>  </span>
<span id="cb38-33"><a href="#cb38-33"></a>  <span class="fu">do.call</span>(rbind, confusion)</span>
<span id="cb38-34"><a href="#cb38-34"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data-simulation-6" class="slide level2 extra">
<h2>Data Simulation</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a><span class="fu">classify</span>(dat, y, x, <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  tp fp tn fn tpr  fpr  tnr fnr tot prevalence c       ppv    npv
1 35 17 33 15 0.7 0.34 0.66 0.3 100        0.5 0 0.6730769 0.6875</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a><span class="fu">classify</span>(dat, y, x, <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  tp fp tn fn  tpr  fpr  tnr  fnr tot prevalence  c       ppv       npv
1 47 36 14  3 0.94 0.72 0.28 0.06 100        0.5 -1 0.5662651 0.8235294
2 35 17 33 15 0.70 0.34 0.66 0.30 100        0.5  0 0.6730769 0.6875000
3 15  3 47 35 0.30 0.06 0.94 0.70 100        0.5  1 0.8333333 0.5731707</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a><span class="co"># ~ full roc curve</span></span>
<span id="cb43-2"><a href="#cb43-2"></a>cs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="cn">Inf</span>, <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">4</span>, <span class="fl">0.1</span>), <span class="cn">Inf</span>)</span>
<span id="cb43-3"><a href="#cb43-3"></a>car<span class="sc">::</span><span class="fu">some</span>(<span class="fu">classify</span>(dat, y, x, cs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   tp fp tn fn  tpr  fpr  tnr  fnr tot prevalence    c       ppv       npv
12 50 50  0  0 1.00 1.00 0.00 0.00 100        0.5 -3.0 0.5000000       NaN
18 50 49  1  0 1.00 0.98 0.02 0.00 100        0.5 -2.4 0.5050505 1.0000000
25 49 45  5  1 0.98 0.90 0.10 0.02 100        0.5 -1.7 0.5212766 0.8333333
27 49 42  8  1 0.98 0.84 0.16 0.02 100        0.5 -1.5 0.5384615 0.8888889
43 31 16 34 19 0.62 0.32 0.68 0.38 100        0.5  0.1 0.6595745 0.6415094
46 26 10 40 24 0.52 0.20 0.80 0.48 100        0.5  0.4 0.7222222 0.6250000
54 12  1 49 38 0.24 0.02 0.98 0.76 100        0.5  1.2 0.9230769 0.5632184
56  8  0 50 42 0.16 0.00 1.00 0.84 100        0.5  1.4 1.0000000 0.5434783
64  3  0 50 47 0.06 0.00 1.00 0.94 100        0.5  2.2 1.0000000 0.5154639
80  0  0 50 50 0.00 0.00 1.00 1.00 100        0.5  3.8       NaN 0.5000000</code></pre>
</div>
</div>
</section>
<section id="data-simulation-7" class="slide level2 extra">
<h2>Data Simulation</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a>cr <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>, <span class="fl">0.1</span>) <span class="co"># vector of thresholds</span></span>
<span id="cb45-2"><a href="#cb45-2"></a>cmat <span class="ot">&lt;-</span> <span class="fu">classify</span>(dat, y, x, cr)</span>
<span id="cb45-3"><a href="#cb45-3"></a></span>
<span id="cb45-4"><a href="#cb45-4"></a><span class="fu">plot</span>(cmat<span class="sc">$</span>tnr, cmat<span class="sc">$</span>tpr, <span class="at">xlim =</span> <span class="fu">rev</span>(<span class="fu">range</span>(cmat<span class="sc">$</span>tpr)), <span class="at">type =</span> <span class="st">"l"</span>,</span>
<span id="cb45-5"><a href="#cb45-5"></a>     <span class="at">xlab =</span> <span class="st">"Specificity"</span>,</span>
<span id="cb45-6"><a href="#cb45-6"></a>     <span class="at">ylab =</span> <span class="st">"Sensitivity"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-54-1.svg" width="960" class="r-stretch quarto-figure-center"></section></section>
<section>
<section id="application-to-genetic-data" class="title-slide slide level1 section center">
<h1>Application to genetic data</h1>

</section>
<section id="objectives" class="slide level2">
<h2>Objectives</h2>
<p>Moving to our <code>curatedOvarianData</code> dataset, we want to explore how the information from genes (<span class="math inline">\(g\)</span>) expression predict the <strong>debulking</strong> probability.</p>
<blockquote>
<p>The term <strong>debulking</strong> represent the patient‚Äôs surgery success in removing the tumor mass (<em>optimal debulking</em>) vs the insuccess in removing the tumor mass (<em>suboptimal debulking</em>)</p>
</blockquote>
<p>Some questions are:</p>
<ul>
<li>how different genes predict the <strong>debulking</strong> status</li>
<li>how results from a given gene <span class="math inline">\(g_i\)</span> change, using a different dataset</li>
</ul>
</section>
<section id="conventions-and-notation" class="slide level2">
<h2>Conventions and notation</h2>
<p>We saw that working with contingency tables requires giving labels to the actual data. In this case we set the following conventions:</p>
<ul>
<li>the <strong>suboptimal</strong> state is coded as 0 i.e.&nbsp;having the <em>disease</em></li>
<li>the <strong>optimal</strong> state is coded as 1 i.e.&nbsp;not having the <em>disease</em></li>
<li>higher values of gene expressions (<span class="math inline">\(x_{g_i}\)</span>) are associated with higher probability of <strong>suboptimal</strong> state</li>
<li>we call the <strong>state</strong> (S) the variable representing the <em>TRUE</em> surgery outcome and <strong>test</strong> (T) the predicted outcome using the gene expression. Thus e.g.&nbsp;<span class="math inline">\(p(T_1|S_1)\)</span> is the Sensitivity</li>
</ul>
</section>
<section id="steps" class="slide level2">
<h2>Steps</h2>
<ol type="1">
<li>Load and explore the data (you can the <code>gene_data_clean.rds</code> or using the raw <code>curatedOvarianData</code>)</li>
<li>Calculate the prevalence (<span class="math inline">\(p(S_1)\)</span>) of the <strong>optimal</strong> debulking for each dataset</li>
<li>Choose a gene (<span class="math inline">\(g_i\)</span>) and plot the <span class="math inline">\(x_i\)</span> values according to the debulking</li>
<li>Calculate the Sensitivity, Specificity, Youden‚Äôs J and PPV for all datasets based on the same gene.</li>
<li>Compare graphically and descriptively the results</li>
<li>Calculate the standard error of the AUC (see the <code>pROC::var()</code> function)</li>
<li>Compute and represent a <strong>random-effects meta-analysis</strong> polling the AUC for the same gene, across datasets. Interpret the pooled effect, heterogeneity and Q-statistics.</li>
</ol>
</section>
<section id="steps-optional" class="slide level2">
<h2>Steps (optional)</h2>
<ol start="6" type="1">
<li>[Extra] Try to compute the standard error of the AUC using a <em>bootstrapping</em> approach. Essentially you need to sample with replacement (see <code>sample()</code>) <span class="math inline">\(n\)</span> rows from the dataset (where <span class="math inline">\(n\)</span> is the total number of rows), compute and store the AUC, repeat the process several times (at least 10000 times), calculate the standard deviation of the boostrapped AUC distribution.</li>
</ol>
<ul>
<li>There is also the <code>pROC::ci.auc()</code> function with the <code>bootstrap</code> option. Try to manually code the boostrap and eventually compare the results</li>
<li>The confidence interval can be calculated with the <code>quantile()</code> function <span class="citation" data-cites="Carpenter2000-dm">(see <a href="#/references" role="doc-biblioref" onclick="">Carpenter and Bithell 2000</a>)</span> with the desired coverage (e.g., 95%)</li>
<li>Further details are available here https://cran.r-project.org/web/packages/pROC/pROC.pdf (page 5, Section Bootstrap)</li>
</ul>
</section></section>
<section>
<section id="proposed-solutions" class="title-slide slide level1 section center">
<h1>Proposed solutions</h1>

</section>
<section id="setup" class="slide level2">
<h2>Setup</h2>
<p>Firstly, let‚Äôs subset the dataset selecting our predictor (i.e., the <strong>debulking</strong>), the dataset name and the selected gene.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a>datl <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"03-biostatistics"</span>, <span class="st">"objects"</span>, <span class="st">"gene_data_clean.rds"</span>))</span>
<span id="cb46-2"><a href="#cb46-2"></a>dat <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(datl, <span class="at">.id =</span> <span class="st">"dataset"</span>)</span>
<span id="cb46-3"><a href="#cb46-3"></a></span>
<span id="cb46-4"><a href="#cb46-4"></a><span class="co"># creating the binary debulking variable</span></span>
<span id="cb46-5"><a href="#cb46-5"></a>dat<span class="sc">$</span>debulking01 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(dat<span class="sc">$</span>debulking <span class="sc">==</span> <span class="st">"optimal"</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb46-6"><a href="#cb46-6"></a></span>
<span id="cb46-7"><a href="#cb46-7"></a>gene <span class="ot">&lt;-</span> <span class="st">"ABCA1"</span></span>
<span id="cb46-8"><a href="#cb46-8"></a>dat_meta <span class="ot">&lt;-</span> dat[, <span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"debulking"</span>, gene)]</span>
<span id="cb46-9"><a href="#cb46-9"></a><span class="fu">head</span>(dat_meta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          dataset  debulking     ABCA1
1 E.MTAB.386_eset suboptimal 10.839395
2 E.MTAB.386_eset    optimal 11.227465
3 E.MTAB.386_eset    optimal 11.375979
4 E.MTAB.386_eset    optimal  9.716625
5 E.MTAB.386_eset suboptimal 11.647426
6 E.MTAB.386_eset    optimal 10.291002</code></pre>
</div>
</div>
</section>
<section id="prevalence" class="slide level2">
<h2>Prevalence</h2>
<p>Now let‚Äôs see which is the prevalence of <em>optimal</em> debulking i.e.&nbsp;<span class="math inline">\(S_1\)</span></p>
<div class="panel-tabset">
<ul id="tabset-4" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-4-1">Code</a></li><li><a href="#tabset-4-2">Plot</a></li></ul>
<div class="tab-content">
<div id="tabset-4-1">
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a>prev_and_n <span class="ot">&lt;-</span> <span class="cf">function</span>(data){</span>
<span id="cb48-2"><a href="#cb48-2"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb48-3"><a href="#cb48-3"></a>    <span class="at">p =</span> <span class="fu">mean</span>(data<span class="sc">$</span>debulking01, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb48-4"><a href="#cb48-4"></a>    <span class="at">n =</span> <span class="fu">nrow</span>(data)</span>
<span id="cb48-5"><a href="#cb48-5"></a>  )</span>
<span id="cb48-6"><a href="#cb48-6"></a>}</span>
<span id="cb48-7"><a href="#cb48-7"></a>prev <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">split</span>(dat_meta, dat_meta<span class="sc">$</span>dataset), prev_and_n)</span>
<span id="cb48-8"><a href="#cb48-8"></a>prev <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(prev, <span class="at">.id =</span> <span class="st">"dataset"</span>)</span>
<span id="cb48-9"><a href="#cb48-9"></a><span class="fu">head</span>(prev)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          dataset  p   n
1 E.MTAB.386_eset NA 129
2   GSE12418_eset NA  54
3   GSE17260_eset NA 110
4   GSE26712_eset NA 195
5   GSE30009_eset NA 103
6   GSE30161_eset NA  58</code></pre>
</div>
</div>
</div>
<div id="tabset-4-2">
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1"></a>prev <span class="sc">|&gt;</span> </span>
<span id="cb50-2"><a href="#cb50-2"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> p<span class="sc">*</span><span class="dv">100</span>, <span class="at">y =</span> dataset)) <span class="sc">+</span></span>
<span id="cb50-3"><a href="#cb50-3"></a>  <span class="fu">geom_col</span>(<span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb50-4"><a href="#cb50-4"></a>  <span class="fu">xlab</span>(<span class="st">"Prevalence (%)"</span>) <span class="sc">+</span></span>
<span id="cb50-5"><a href="#cb50-5"></a>  <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-57-1.svg" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="debulking-gene" class="slide level2">
<h2>debulking ~ gene</h2>
<p>Let‚Äôs see the density plots for dataset with the gene ABCA1:</p>
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>dat_meta <span class="sc">|&gt;</span> </span>
<span id="cb51-2"><a href="#cb51-2"></a>  <span class="fu">drop_na</span>(debulking) <span class="sc">|&gt;</span> </span>
<span id="cb51-3"><a href="#cb51-3"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> ABCA1, <span class="at">fill =</span> debulking)) <span class="sc">+</span></span>
<span id="cb51-4"><a href="#cb51-4"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb51-5"><a href="#cb51-5"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb51-6"><a href="#cb51-6"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="fl">0.1</span>),</span>
<span id="cb51-7"><a href="#cb51-7"></a>        <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb51-8"><a href="#cb51-8"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-58-1.svg" width="960" class="r-stretch quarto-figure-center"><p>In some datasets, the gene ABCA1 discriminate more between the the debulking outcomes.</p>
</section>
<section id="auc-and-standard-error" class="slide level2">
<h2>AUC and standard error</h2>
<div class="panel-tabset">
<ul id="tabset-5" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-5-1">ROC</a></li><li><a href="#tabset-5-2">Analytical method</a></li><li><a href="#tabset-5-3">Plot</a></li><li><a href="#tabset-5-4">Bootstrapping</a></li><li><a href="#tabset-5-5">Plot</a></li></ul>
<div class="tab-content">
<div id="tabset-5-1">
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1"></a>roc_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">split</span>(dat_meta, dat_meta<span class="sc">$</span>dataset), </span>
<span id="cb52-2"><a href="#cb52-2"></a>                   <span class="cf">function</span>(x) pROC<span class="sc">::</span><span class="fu">roc</span>(debulking <span class="sc">~</span> ABCA1, <span class="at">data =</span> x))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
<div id="tabset-5-2">
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a>aucs <span class="ot">&lt;-</span> <span class="fu">sapply</span>(roc_list,  auc)</span>
<span id="cb53-2"><a href="#cb53-2"></a><span class="co"># the var function is used to compute the sampling variability</span></span>
<span id="cb53-3"><a href="#cb53-3"></a>aucs_se <span class="ot">&lt;-</span> <span class="fu">sapply</span>(roc_list, <span class="cf">function</span>(x) <span class="fu">sqrt</span>(<span class="fu">var</span>(x)))</span>
<span id="cb53-4"><a href="#cb53-4"></a>auc_d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">dataset =</span> <span class="fu">names</span>(aucs), <span class="at">auc =</span> aucs, <span class="at">se =</span> aucs_se)</span>
<span id="cb53-5"><a href="#cb53-5"></a>ci <span class="ot">&lt;-</span> <span class="fu">sapply</span>(roc_list, <span class="cf">function</span>(x) pROC<span class="sc">::</span><span class="fu">ci.auc</span>(x))</span>
<span id="cb53-6"><a href="#cb53-6"></a>auc_d<span class="sc">$</span>lower <span class="ot">&lt;-</span> ci[<span class="dv">1</span>, ]</span>
<span id="cb53-7"><a href="#cb53-7"></a>auc_d<span class="sc">$</span>upper <span class="ot">&lt;-</span> ci[<span class="dv">3</span>, ]</span>
<span id="cb53-8"><a href="#cb53-8"></a></span>
<span id="cb53-9"><a href="#cb53-9"></a><span class="fu">head</span>(auc_d)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                        dataset       auc         se     lower     upper
E.MTAB.386_eset E.MTAB.386_eset 0.4713921 0.07114057 0.3319592 0.6108251
GSE12418_eset     GSE12418_eset 0.4969136 0.10737807 0.2864564 0.7073707
GSE17260_eset     GSE17260_eset 0.5160543 0.05577309 0.4067410 0.6253675
GSE26712_eset     GSE26712_eset 0.4933333 0.04290461 0.4092418 0.5774248
GSE30009_eset     GSE30009_eset 0.5263749 0.07303915 0.3832208 0.6695290
GSE30161_eset     GSE30161_eset 0.5435897 0.08069339 0.3854336 0.7017459</code></pre>
</div>
</div>
</div>
<div id="tabset-5-3">
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a>auc_d <span class="sc">|&gt;</span> </span>
<span id="cb55-2"><a href="#cb55-2"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> auc, <span class="at">y =</span> dataset)) <span class="sc">+</span></span>
<span id="cb55-3"><a href="#cb55-3"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb55-4"><a href="#cb55-4"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> lower, <span class="at">xend =</span> upper, <span class="at">y =</span> dataset, <span class="at">yend =</span> dataset)) <span class="sc">+</span></span>
<span id="cb55-5"><a href="#cb55-5"></a>  <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-61-1.svg" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
<div id="tabset-5-4">
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1"></a>boot_auc <span class="ot">&lt;-</span> <span class="cf">function</span>(data, <span class="at">direction =</span> <span class="st">"auto"</span>, <span class="at">nboot =</span> <span class="dv">10</span>){</span>
<span id="cb56-2"><a href="#cb56-2"></a>  aucs <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, nboot)</span>
<span id="cb56-3"><a href="#cb56-3"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nboot){</span>
<span id="cb56-4"><a href="#cb56-4"></a>    idx <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data), <span class="fu">nrow</span>(data), <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb56-5"><a href="#cb56-5"></a>    datb <span class="ot">&lt;-</span> data[idx, ]</span>
<span id="cb56-6"><a href="#cb56-6"></a>    aucs[[i]] <span class="ot">&lt;-</span> pROC<span class="sc">::</span><span class="fu">auc</span>(datb[, <span class="dv">2</span>] <span class="sc">~</span> datb[, <span class="dv">3</span>], <span class="at">direction =</span> direction, <span class="at">quiet =</span> <span class="cn">TRUE</span>)[[<span class="dv">1</span>]]</span>
<span id="cb56-7"><a href="#cb56-7"></a>  }</span>
<span id="cb56-8"><a href="#cb56-8"></a>  <span class="fu">return</span>(aucs)</span>
<span id="cb56-9"><a href="#cb56-9"></a>}</span>
<span id="cb56-10"><a href="#cb56-10"></a></span>
<span id="cb56-11"><a href="#cb56-11"></a><span class="co"># get the direction, auto detecting direction could lead to biased AUC see https://cran.r-project.org/web/packages/pROC/pROC.pdf</span></span>
<span id="cb56-12"><a href="#cb56-12"></a>direction <span class="ot">&lt;-</span> <span class="fu">sapply</span>(roc_list, <span class="cf">function</span>(x) x<span class="sc">$</span>direction)</span>
<span id="cb56-13"><a href="#cb56-13"></a></span>
<span id="cb56-14"><a href="#cb56-14"></a>auc_list <span class="ot">&lt;-</span> <span class="fu">mapply</span>(<span class="cf">function</span>(d, dir) <span class="fu">boot_auc</span>(d, dir), <span class="fu">split</span>(dat_meta, dat_meta<span class="sc">$</span>dataset), direction)</span>
<span id="cb56-15"><a href="#cb56-15"></a>aucs_m <span class="ot">&lt;-</span>  <span class="fu">apply</span>(auc_list, <span class="dv">2</span>, median)</span>
<span id="cb56-16"><a href="#cb56-16"></a>aucs_se <span class="ot">&lt;-</span> <span class="fu">apply</span>(auc_list, <span class="dv">2</span>, sd)</span>
<span id="cb56-17"><a href="#cb56-17"></a>aucs_ci <span class="ot">&lt;-</span> <span class="fu">apply</span>(auc_list, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">quantile</span>(x, <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.975</span>)))</span>
<span id="cb56-18"><a href="#cb56-18"></a>aucs_d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">auc =</span> aucs_m, <span class="at">se =</span> aucs_se, <span class="at">lower =</span> aucs_ci[<span class="dv">1</span>, ], <span class="at">upper =</span> aucs_ci[<span class="dv">2</span>, ], <span class="at">dataset =</span> <span class="fu">names</span>(roc_list))</span>
<span id="cb56-19"><a href="#cb56-19"></a></span>
<span id="cb56-20"><a href="#cb56-20"></a><span class="fu">head</span>(aucs_d)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      auc         se     lower     upper         dataset
E.MTAB.386_eset 0.5044379 0.06213286 0.3842916 0.5564459 E.MTAB.386_eset
GSE12418_eset   0.4696637 0.09483494 0.3720486 0.6545699   GSE12418_eset
GSE17260_eset   0.5622395 0.04842584 0.4555081 0.6045954   GSE17260_eset
GSE26712_eset   0.4916090 0.03651884 0.4401318 0.5503618   GSE26712_eset
GSE30009_eset   0.5604650 0.07183862 0.5104508 0.7084291   GSE30009_eset
GSE30161_eset   0.5250189 0.06744519 0.4489217 0.6277099   GSE30161_eset</code></pre>
</div>
</div>
</div>
<div id="tabset-5-5">
<div class="cell" data-layout-align="center">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1"></a>aucs_d <span class="sc">|&gt;</span> </span>
<span id="cb58-2"><a href="#cb58-2"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> auc, <span class="at">y =</span> dataset)) <span class="sc">+</span></span>
<span id="cb58-3"><a href="#cb58-3"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb58-4"><a href="#cb58-4"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> lower, <span class="at">xend =</span> upper, <span class="at">y =</span> dataset, <span class="at">yend =</span> dataset)) <span class="sc">+</span></span>
<span id="cb58-5"><a href="#cb58-5"></a>  <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-63-1.svg" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="meta-analysis" class="slide level2">
<h2>Meta-analysis</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1"></a><span class="fu">library</span>(metafor)</span>
<span id="cb59-2"><a href="#cb59-2"></a>auc_d<span class="sc">$</span>var <span class="ot">&lt;-</span> auc_d<span class="sc">$</span>se<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb59-3"><a href="#cb59-3"></a></span>
<span id="cb59-4"><a href="#cb59-4"></a>fit_rma <span class="ot">&lt;-</span> <span class="fu">rma</span>(auc, var, <span class="at">data =</span> auc_d)</span>
<span id="cb59-5"><a href="#cb59-5"></a><span class="fu">summary</span>(fit_rma)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Random-Effects Model (k = 15; tau^2 estimator: REML)

  logLik  deviance       AIC       BIC      AICc   
 20.1608  -40.3216  -36.3216  -35.0435  -35.2307   

tau^2 (estimated amount of total heterogeneity): 0.0001 (SE = 0.0009)
tau (square root of estimated tau^2 value):      0.0098
I^2 (total heterogeneity / total variability):   3.50%
H^2 (total variability / sampling variability):  1.04

Test for Heterogeneity:
Q(df = 14) = 13.0511, p-val = 0.5225

Model Results:

estimate      se     zval    pval   ci.lb   ci.ub      
  0.5597  0.0134  41.7881  &lt;.0001  0.5334  0.5859  *** 

---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</section>
<section id="meta-analysis-1" class="slide level2">
<h2>Meta-analysis</h2>
<p>We need to adjust the test statistics because the null value with the AUC is not 0 but 0.5.</p>
<p><span class="math display">\[
z = \frac{\hat{AUC} - AUC_0}{\hat{SE_{AUC}}}
\]</span></p>
<p><span class="math display">\[
z = \frac{\hat{AUC} - 0.5}{SE_{\hat{AUC}}}
\]</span></p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1"></a><span class="co"># pvalue testing the null of 0.5</span></span>
<span id="cb61-2"><a href="#cb61-2"></a><span class="fu">pnorm</span>((fit_rma<span class="sc">$</span>b[[<span class="dv">1</span>]] <span class="sc">-</span> <span class="fl">0.5</span>)<span class="sc">/</span>fit_rma<span class="sc">$</span>se, <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)<span class="sc">*</span><span class="dv">2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 8.38819e-06</code></pre>
</div>
</div>
</section>
<section id="forest-plot" class="slide level2">
<h2>Forest Plot</h2>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1"></a><span class="fu">forest</span>(fit_rma, <span class="at">refline =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="03-biostatistics_files/figure-revealjs/unnamed-chunk-66-1.svg" width="960" class="r-stretch quarto-figure-center"></section></section>
<section>
<section id="comments" class="title-slide slide level1 question center">
<h1>Comments?</h1>

</section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<a onclick="fetch('data:text/x-bibtex;base64,@UNPUBLISHED{Parmigiani2023-dr,
  title = {Defining replicability of prediction rules},
  author = {Parmigiani, Giovanni},
  journaltitle = {arXiv [stat.ME]},
  date = {2023-04-30},
  eprint = {2305.01518},
  eprintclass = {stat.ME},
  abstract = {In this article I propose an approach for defining replicability
  for prediction rules. Motivated by a recent NAS report, I start from the
  perspective that replicability is obtaining consistent results across studies
  suitable to address the same prediction question, each of which has obtained
  its own data. I then discuss concept and issues in defining key elements of
  this statement. I focus specifically on the meaning of "consistent results" in
  typical utilization contexts, and propose a multi-agent framework for defining
  replicability, in which agents are neither partners nor adversaries. I recover
  some of the prevalent practical approaches as special cases. I hope to provide
  guidance for a more systematic assessment of replicability in machine
  learning.},
  url = {http://arxiv.org/abs/2305.01518},
  urldate = {2023-08-22}
}

@ARTICLE{Brenner1997-yq,
  title = {Variation of sensitivity, specificity, likelihood ratios and
  predictive values with disease prevalence},
  author = {Brenner, H and Gefeller, O},
  journaltitle = {Stat. Med.},
  publisher = {Wiley},
  volume = {16},
  issue = {9},
  pages = {981-991},
  date = {1997-05-15},
  doi = {10.1002/(sici)1097-0258(19970515)16:9<981::aid-sim510>3.0.co;2-n},
  pmid = {9160493},
  issn = {1097-0258,0277-6715},
  abstract = {The sensitivity, specificity and likelihood ratios of binary
  diagnostic tests are often thought of as being independent of disease
  prevalence. Empirical studies, however, have frequently revealed substantial
  variation of these measures for the same diagnostic test in different
  populations. One reason for this discrepancy is related to the fact that only
  few diagnostic tests are inherently dichotomous. The majority of tests are
  based on categorization of individuals according to one or several underlying
  continuous traits. For these tests, the magnitude of diagnostic
  misclassification depends not only on the magnitude of the measurement or
  perception error of the underlying trait(s), but also on the distribution of
  the underlying trait(s) in the population relative to the diagnostic cutpoint.
  Since this distribution also determines prevalence of the disease in the
  population, diagnostic misclassification and disease prevalence are related
  for this type of test. We assess the variation of various measures of validity
  of diagnostic tests with disease prevalence for simple models of the
  distribution of the underlying trait(s) and the measurement or perception
  error. We illustrate that variation with disease prevalence is typically
  strong for sensitivity and specificity, and even more so for the likelihood
  ratios. Although positive and negative predictive values also strongly vary
  with disease prevalence, this variation is usually less pronounced than one
  would expect if sensitivity and specificity were independent of disease
  prevalence.},
  url = {https://onlinelibrary.wiley.com/doi/abs/10.1002/%28SICI%291097-0258%2819970515%2916%3A9%3C981%3A%3AAID-SIM510%3E3.0.CO%3B2-N},
  urldate = {2023-08-18},
  language = {en}
}

@ARTICLE{Carpenter2000-dm,
  title = {Bootstrap confidence intervals: when, which, what? A practical guide
  for medical statisticians},
  author = {Carpenter, J and Bithell, J},
  journaltitle = {Stat. Med.},
  publisher = {Wiley},
  volume = {19},
  issue = {9},
  pages = {1141-1164},
  date = {2000-05-15},
  doi = {10.1002/(sici)1097-0258(20000515)19:9<1141::aid-sim479>3.0.co;2-f},
  pmid = {10797513},
  issn = {1097-0258,0277-6715},
  abstract = {Since the early 1980s, a bewildering array of methods for
  constructing bootstrap confidence intervals have been proposed. In this
  article, we address the following questions. First, when should bootstrap
  confidence intervals be used. Secondly, which method should be chosen, and
  thirdly, how should it be implemented. In order to do this, we review the
  common algorithms for resampling and methods for constructing bootstrap
  confidence intervals, together with some less well known ones, highlighting
  their strengths and weaknesses. We then present a simulation study, a flow
  chart for choosing an appropriate method and a survival analysis example.},
  url = {https://onlinelibrary.wiley.com/doi/abs/10.1002/%28SICI%291097-0258%2820000515%2919%3A9%3C1141%3A%3AAID-SIM479%3E3.0.CO%3B2-F},
  urldate = {2023-08-23},
  language = {en}
}

@ARTICLE{Patil2018-hm,
  title = {Training replicable predictors in multiple studies},
  author = {Patil, Prasad and Parmigiani, Giovanni},
  journaltitle = {Proc. Natl. Acad. Sci. U. S. A.},
  publisher = {National Academy of Sciences},
  volume = {115},
  issue = {11},
  pages = {2578-2583},
  date = {2018-03-13},
  doi = {10.1073/pnas.1708283115},
  pmc = {PMC5856504},
  pmid = {29531060},
  issn = {0027-8424,1091-6490},
  abstract = {This article considers replicability of the performance of
  predictors across studies. We suggest a general approach to investigating this
  issue, based on ensembles of prediction models trained on different studies.
  We quantify how the common practice of training on a single study accounts in
  part for the observed challenges in replicability of prediction performance.
  We also investigate whether ensembles of predictors trained on multiple
  studies can be combined, using unique criteria, to design robust ensemble
  learners trained upfront to incorporate replicability into different contexts
  and populations.},
  url = {https://www.pnas.org/doi/abs/10.1073/pnas.1708283115},
  urldate = {2023-08-22},
  keywords = {cross-study validation; ensemble learning; machine learning;
  replicability; validation},
  language = {en}
}

@ARTICLE{Ganzfried2013-ut,
  title = {curatedOvarianData: clinically annotated data for the ovarian cancer
  transcriptome},
  author = {Ganzfried, Benjamin Frederick and Riester, Markus and Haibe-Kains,
  Benjamin and Risch, Thomas and Tyekucheva, Svitlana and Jazic, Ina and Wang,
  Xin Victoria and Ahmadifar, Mahnaz and Birrer, Michael J and Parmigiani,
  Giovanni and Huttenhower, Curtis and Waldron, Levi},
  journaltitle = {Database (Oxford)},
  publisher = {Oxford University Press (OUP)},
  volume = {2013},
  pages = {bat013},
  date = {2013-04-02},
  doi = {10.1093/database/bat013},
  pmc = {PMC3625954},
  pmid = {23550061},
  issn = {1758-0463},
  abstract = {This article introduces a manually curated data collection for
  gene expression meta-analysis of patients with ovarian cancer and software for
  reproducible preparation of similar databases. This resource provides
  uniformly prepared microarray data for 2970 patients from 23 studies with
  curated and documented clinical metadata. It allows users to efficiently
  identify studies and patient subgroups of interest for analysis and to perform
  meta-analysis immediately without the challenges posed by harmonizing
  heterogeneous microarray technologies, study designs, expression data
  processing methods and clinical data formats. We confirm that the recently
  proposed biomarker CXCL12 is associated with patient survival, independently
  of stage and optimal surgical debulking, which was possible only through
  meta-analysis owing to insufficient sample sizes of the individual studies.
  The database is implemented as the curatedOvarianData Bioconductor package for
  the R statistical computing language, providing a comprehensive and flexible
  resource for clinically oriented investigation of the ovarian cancer
  transcriptome. The package and pipeline for producing it are available from
  http://bcb.dfci.harvard.edu/ovariancancer.},
  url = {https://academic.oup.com/database/article/doi/10.1093/database/bat013/330978?view=long},
  language = {en}
}

@ARTICLE{Masoero2023-wb,
  title = {Cross-Study Replicability in Cluster Analysis},
  author = {Masoero, Lorenzo and Thomas, Emma and Parmigiani, Giovanni and
  Tyekucheva, Svitlana and Trippa, Lorenzo},
  journaltitle = {Stat. Sci.},
  publisher = {Institute of Mathematical Statistics},
  volume = {38},
  issue = {2},
  pages = {303-316},
  date = {2023-05-01},
  doi = {10.1214/22-sts871},
  issn = {0883-4237,2168-8745},
  abstract = {In cancer research, clustering techniques are widely used for
  exploratory analyses, playing a critical role in the identification of novel
  cancer subtypes and patient management. As data collected by multiple research
  groups grows, it is increasingly feasible to investigate the replicability of
  clustering procedures, that is, their ability to consistently recover
  biologically meaningful clusters across several data sets. In this paper, we
  review methods for replicability of clustering analyses, and discuss a novel
  framework for evaluating cross-study clustering replicability, useful when two
  or more studies are available. Our approach can be applied to any clustering
  algorithm and can employ different measures of similarity between partitions
  to quantify replicability, globally (i.e., for the whole sample) as well as
  locally (i.e., for individual clusters). Using experiments on synthetic and
  real gene expression data, we illustrate the usefulness of our procedure to
  evaluate if the same clusters are identified consistently across a collection
  of data sets.},
  url = {https://projecteuclid.org/journals/statistical-science/volume-38/issue-2/Cross-Study-Replicability-in-Cluster-Analysis/10.1214/22-STS871.short},
  urldate = {2023-08-22},
  keywords = {clustering; multiple studies; replicability},
  language = {en}
}

').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'references.bib'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
    });" href="">
<button class="btn btn-danger"><i class="fa fa-book"></i> Download the bibtex file</button>
</a>
</div>
</div>
<p><br></p>
<div class="footer footer-default">

</div>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-Brenner1997-yq" class="csl-entry" role="listitem">
Brenner, H, and O Gefeller. 1997. <span>‚ÄúVariation of Sensitivity, Specificity, Likelihood Ratios and Predictive Values with Disease Prevalence.‚Äù</span> <em>Stat. Med.</em> 16 (May): 981‚Äì91. <a href="https://doi.org/10.1002/(sici)1097-0258(19970515)16:9<981::aid-sim510>3.0.co;2-n">https://doi.org/10.1002/(sici)1097-0258(19970515)16:9&lt;981::aid-sim510&gt;3.0.co;2-n</a>.
</div>
<div id="ref-Carpenter2000-dm" class="csl-entry" role="listitem">
Carpenter, J, and J Bithell. 2000. <span>‚ÄúBootstrap Confidence Intervals: When, Which, What? A Practical Guide for Medical Statisticians.‚Äù</span> <em>Stat. Med.</em> 19 (May): 1141‚Äì64. <a href="https://doi.org/10.1002/(sici)1097-0258(20000515)19:9<1141::aid-sim479>3.0.co;2-f">https://doi.org/10.1002/(sici)1097-0258(20000515)19:9&lt;1141::aid-sim479&gt;3.0.co;2-f</a>.
</div>
<div id="ref-Ganzfried2013-ut" class="csl-entry" role="listitem">
Ganzfried, Benjamin Frederick, Markus Riester, Benjamin Haibe-Kains, Thomas Risch, Svitlana Tyekucheva, Ina Jazic, Xin Victoria Wang, et al. 2013. <span>‚ÄúcuratedOvarianData: Clinically Annotated Data for the Ovarian Cancer Transcriptome.‚Äù</span> <em>Database (Oxford)</em> 2013 (April): bat013. <a href="https://doi.org/10.1093/database/bat013">https://doi.org/10.1093/database/bat013</a>.
</div>
<div id="ref-Masoero2023-wb" class="csl-entry" role="listitem">
Masoero, Lorenzo, Emma Thomas, Giovanni Parmigiani, Svitlana Tyekucheva, and Lorenzo Trippa. 2023. <span>‚ÄúCross-Study Replicability in Cluster Analysis.‚Äù</span> <em>Stat. Sci.</em> 38 (May): 303‚Äì16. <a href="https://doi.org/10.1214/22-sts871">https://doi.org/10.1214/22-sts871</a>.
</div>
<div id="ref-Parmigiani2023-dr" class="csl-entry" role="listitem">
Parmigiani, Giovanni. 2023. <span>‚ÄúDefining Replicability of Prediction Rules.‚Äù</span> <em>arXiv [Stat.ME]</em>. <a href="http://arxiv.org/abs/2305.01518">http://arxiv.org/abs/2305.01518</a>.
</div>
<div id="ref-Patil2018-hm" class="csl-entry" role="listitem">
Patil, Prasad, and Giovanni Parmigiani. 2018. <span>‚ÄúTraining Replicable Predictors in Multiple Studies.‚Äù</span> <em>Proc. Natl. Acad. Sci. U. S. A.</em> 115 (March): 2578‚Äì83. <a href="https://doi.org/10.1073/pnas.1708283115">https://doi.org/10.1073/pnas.1708283115</a>.
</div>
<div id="ref-Waldron2014-yv" class="csl-entry" role="listitem">
Waldron, Levi, Benjamin Haibe-Kains, Aed√≠n C Culhane, Markus Riester, Jie Ding, Xin Victoria Wang, Mahnaz Ahmadifar, et al. 2014. <span>‚ÄúComparative Meta-Analysis of Prognostic Gene Signatures for Late-Stage Ovarian Cancer.‚Äù</span> <em>J. Natl. Cancer Inst.</em> 106 (April): dju049. <a href="https://doi.org/10.1093/jnci/dju049">https://doi.org/10.1093/jnci/dju049</a>.
</div>
</div>
</section></section>

<!-- FINAL SLIDE -->


<!-- END FINAL SLIDE -->

    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="03-biostatistics_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="03-biostatistics_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="03-biostatistics_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="03-biostatistics_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="03-biostatistics_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="03-biostatistics_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="03-biostatistics_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="03-biostatistics_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="03-biostatistics_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="03-biostatistics_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: false,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    
</body></html>